---
title: "**Antibody Response Induced by HIV Vaccines and T-cell Suppression Treatments in Rhesus Macaques**"
author: "Group 3: Kan Luo, Shih-Ni Prim, Frederick Davey, Rizwana Rehman"
date: "11/14/2020"
fontsize: 12pt
geometry: margin = 1in
mainfont: Times New Roman
linestretch: 1.5
fig_caption: yes
indent: true
header-includes:
#  - \usepackage{endfloat}  
  - \usepackage[nomarkers]{endfloat}
output: 
  bookdown::pdf_document2:
    latex_engine: lualatex
    toc: true
    toc_depth: 4
    fig_width: 6.5
    fig_height: 4
    df_print: kable
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, eval = TRUE, message = FALSE, warning = FALSE, cache = TRUE)
library(readxl)
library(tidyverse)
library(ggplot2)
library(httr)
library(nlme)
library(emmeans)
library(GGally)
library(knitr)
library(ICSNP)
library(clubSandwich)
library(bookdown)
```



```{r readin Data, results = 'hide'}
#Readin data from Github url address
url1 <- 'https://github.com/luokan1227/537P1/raw/master/Data.xlsx'  # main data
url2 <- 'https://github.com/luokan1227/537P1/raw/master/MonkeyID.xlsx'  # Monkey ID
GET(url1, write_disk(tf1 <- tempfile(fileext = ".xlsx")))
GET(url2, write_disk(tf2 <- tempfile(fileext = ".xlsx")))
Rawdata <- read_excel(tf1)
MonkeyID <-  read_excel(tf2)
#Add Monkey ID to raw data
Rawdata <- inner_join(Rawdata, MonkeyID, by = "H_ID")
colnames(Rawdata)[2] <- "Time_Point"
#-----------------------
#Adjust or add variables
#-----------------------
# Final data set for all following analysis
Data <- Rawdata
# add drug type and reactivity
Data$Drug <- ifelse(Data$Treatment == "group 7", 3, 
                      ifelse(Data$Treatment == "group 1", 1, 
                      ifelse(Data$Treatment == "group 2", 1, 
                      ifelse(Data$Treatment == "group 3", 1, 2))))
Data$Reactivity <- ifelse(Data$Binding > 0.1, 1, 0)
# create a subset with the variables we need plus the extracted information
Data2 <- Data %>% select(MonkeyID, Drug, Treatment, Time_Point, Isotype, H_VBase, H_Substitutions, H_Insertions, H_Deletions, HMuFreq, H_CDR3, L_VBase, L_Substitutions, L_Insertions, L_Deletions, LMuFreq, L_CDR3, Binding, Reactivity)
```

```{r}
# Introduction
# Conclusion
# Put output at the end
# check assumptions for MANOVA
```


# Introduction  

## About the Study  

A dominant vaccine development strategy is to induce neutralizing antibodies by immunizing humans with the virus' glycoproteins. However, HIV vaccines that adopted this strategy mostly failed due to the fact that HIV is an RNA virus, which mutates rapidly to escape the inhibition of neutralizing antibodies. By the time the body generates neutralizing antibodies against the glycoproteins of some HIV strains, the RNA virus has already mutated. Thus, the existing neutralizing antibody fails to recognize, bind with, and neutralize the HIV virus. One possible solution is to increase the number of potential neutralizing antibodies that will cycle in the body by releasing a variety of antibodies after glycoprotein immunization.  

During the experiment, 20 rhesus macaques were given glycoprotein immunization and supplemental antibody doses, as well as one of three treatments (two experimental regulatory T-cell suppression treatments and one control). For the analysis of mutation frequency and CDR3 count, each antibody within the same treatment was treated as an independent observation.  

Regulatory T (Treg) cells prevent autoimmune diseases and suppress allergic reactions by inhibiting adaptive antibody immune response in the germinal center. Theoretically, this adaptive response lowers the effectiveness of vaccines. Thus the experiment used T-cell suppression treatments to investigate  the effect on immunization. These drugs are widely used in post transplant immuno-suppression treatment to prevent rejection.  

While we might expect different variance within subject vs between subjects, the number of potential antibodies observed is much higher than the number sampled in a blood draw. 

## About the Dataset  

Our dataset includes measurements of antibodies measured in 20 rhesus macaques after they were given the same HIV vaccine at three different time-points and one of three randomly selected anti-Treg treatments(drugs). Blood samples were collected two weeks after vaccine dosing, and antibodies were isolated from those samples. A different number of antibodies were collected from each blood sample, limited by assay yield. Each observation contains information about the antibody isolated post the glycoprotein immunization.  

A human antibody is formed by a heavy chain and light chain. For heavy chain, a human has about 51 V-gene segments, 25 D-gene segments and 6 J-gene segment. For light chain there are 71 V-gene segments and 9 J-gene segments[ref.5]. Any heavy chain V-D-J combination and light chain V-J combinations can randomly happen in germline center. Theoretically, there can be $51*25*6*71*9=$ `r 51*25*6*71*9` combinations of gene segments. Considering the frequently happened mutation and other factors, each individual can have over **10 billion** different antibodies. Thus, we decided to follow the convention of vaccine studies and treat each antibody as independent.   [Kan -- can you identify a reference article or journal here that uses this convention? We don't need to quote it / change the answer, it's just defending our claim that it's a standard practice]  

Below is the list of variables with a brief description from our dataset. Please note that in each antibody, there are two sets of heavy chain and light chain, all of which forming a Y-shape immunoglobulin. Thus many of the variables start with H or L, indicating which chain the information comes from.  

## List of variables  

* Monkey_id: Lists the identity of monkey
* Treatment(Drug): Treatment A is the mock control, and treatment B and C are two different kinds of Treg inhibitor treatments.
* Time_Points: 0 represents before immunization; 1 represents 2 weeks post 1st immunization; 2 represents 2 weeks post 2nd immunization; and 3 represents 2 weeks post 3rd immunization, respectively.
* Isotype: The category of antibody type; there are 5 kinds of immunoglobulin isotypes: IgG, IgA, IgM, IgE, IgD. The two most important kinds are IgG and IgM. IgM occurs in the acute stage of infection and perform an role of primary response. The secondary response IgG appears later in serum with higher binding affinity, and neutralizing potentials against toxins and virus. IgA mostly found in mucosal tissues such as Nasal mucosa. Non-dominant IgD and IgE are typically lower than 1% in blood. 
* H_ID and L_ID: heavy chain and light chain IDs for the particular observation
* H_VBase: the number of nucleotide of the heavy chain variable region
* H_Substitutions: the number of relative nucleotide mutations in heany chain.
* HMuFreq: calculated by H_Substitutions / H_VBase
* H_CDR3: the number of amino acid of the heavy chain's third complementarity determining region 
* L_VBase: the number of nucleotide of the light chain variable region
* L_Substitutions: the number of relative nucleotide mutations in light chain.
* LMuFreq: calculated by L_Substitutions / L_VBase
* L_CDR3: the number of amino acid of the light chain's third complementarity determining region.`H_CDR3` and `L_CDR3` indicates the length of the third complementarity-determining region on the variable heavy chain and light chain. The longer they are, the more potential there is to produce diverse antibodies. [Kan, could you check to see if this is correct?] In other words, we want the values to be higher. 
* Binding: affinity of antibodies against a selected HIV glycoprotein. The larger value indicates stronger binding.`Binding` indicates the rate of neutralizing, meaning how much the antibodies bind with the virus and thus make the virus ineffective. This is the most important measure of the study.   

## Research questions  

The main focus of the current project is to understand whether isotypes, the number of vaccine injections, and the different Treg treatments cause changes in the antibody characteristics and if the changes are related to the immune responses against HIV virus. Specifically, we evaluate:  

**Q1**: Do drugs and isotypes have effects on the mutation frequency and/or the amino acid count in the third complementarity determining region (CDR3)?
  
**Q2**: How does the binding strength of the antibodies develop in response to the number of vaccine dosages by treatment and drug types?   

# Methods  

This section first presents some exploratory data analyses and summaries; then it uses multivariate and longitudinal data analyses to address two research questions.  

## Data Summaries  

### An Overview of Antibodies by Time Points, Drug Types, and Isotypes  

A total of `r nrow(Data2)` antibodies, from 20 rhesus monkeys, were collected at four different time points (0, 1, 2, 3) and each monkey was given one of three drugs (1 and 2 are immuno-suppressing drugs and 3 is the control). Figure \@ref(fig:hist-antibodies) shows the histograms of antibody counts, and Table \@ref(tab:ft-drug-timepoint) and Table \@ref(tab:ft-drug-timepoint-isotype) show the antibody counts in different combinations of  drugs, time points, and isoptypes.  

```{r}
Data2Summary<- Data2 %>% group_by(Drug, Time_Point, MonkeyID) %>% summarize(nObs = n())
Data2Summary$Drug<- as.factor(Data2Summary$Drug)
```

```{r, hist-antibodies, fig.cap = "Histograms of Antibodies"}
# "Histograms of Antibodies Collected per Treatment and Timepoint"
ggplot(data = Data2Summary, mapping = aes(x = nObs, color = Drug)) + geom_histogram(bins = 8, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + labs(xlab = 'n Antibodies Collected', ylab = 'n Monkeys')
```

```{r, ft-drug-timepoint}
knitr::kable(table(Data2$Drug, Data2$Time_Point), row.names = TRUE, caption = "Frequency tables of drug vs. timepoints")
#dnn= c('Drug', 'TimePoint')
```
```{r, ft-drug-timepoint-isotype}
dti <- table(Data2$Drug, Data2$Time_Point, Data2$Isotype, dnn= c('Drug', 'TimePoint', 'Isotype'))
knitr::kable(list(dti[1,,], dti[2,,], dti[3,,]), caption = "Frequency tables of timepoints vs. isotypes for drug = 1 (left), 2 (middle), 3 (right)")
```

```{r, isotype-plot, fig.cap = "Histograms of Isotypes"}
# Histogram 
Iso.order <- c("G", "M", "A", "D", "E") 
g1 <- ggplot(Data, aes(x=factor(Isotype, levels = Iso.order), fill=as.factor(Drug))) 
  g1 + geom_bar(stat = "count", width = .9, position = 'Stack') + # his plot
    facet_grid(cols = vars(Time_Point), labeller = label_both) +
    labs(fill = 'Drug') + # Legend Content
    xlab("Ig Isotype") + ylab("Number of abs") +# X axis and y axis title
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 20) #Title
          #legend.text = element_text(size=20), #Legend text
          #legend.title = element_text(size = 20) # Legend title
    )
```

As shown in Figure \@ref(fig:isotype-plot), the histograms of `Isotype`, we observed that IgG and IgM occupied the biggest proportion of antibodies in all time points. Before immunization (time point 0), there were similar weight of IgG and IgM found in blood. After the first immunization (time point 1), primary immune response resulted an increase of IgM, followed by an IgG increase at later time points 2 and 3.  

### Outlier Detection  

Our response variables for the multivariate analysis include five variables: `Binding`, `H_CDR3`, `HMuFreq`, `L_CDR3`, and `LMuFreq`. As shown in Figure \@ref(fig:outlier-review), `L_CDR3` in one point seems an outlier. As the summary statistics of standardized `L_CDR3` shown in Table \@ref(tab:outlier-tab), the maximum value is greater than 30, which is quite unusal. Figure \@ref(fig:outlier-d2) shows the Mahalanobis ditances and Z scores of `L_CDR3`, and the data point again appears to be an apparent outlier. The value for `L_CDR3` is quite unlikely. Since we can't go back to the original data, we remove the data point and will use the new dataset `Data3`.  

```{r, outlier-review, fig.cap = "Histogram of Response Variables"}
cont_subset <- Data2 %>% select(H_CDR3, HMuFreq, L_CDR3, LMuFreq, Binding)
GGally::ggpairs(cont_subset)
```


```{r, outlier-tab}
standardized <- scale(Data2$L_CDR3)
outlier_tab <- as.table(summary(standardized))
knitr::kable(outlier_tab, caption = "Summaries of standardized LCDR3")
#Summary Statistics for standardized L_CDR3
```

```{r, outlier-d2, fig.cap = "Mahalanobis distances and Z scores"}
subset1 <- Data2 %>% select(L_CDR3, H_CDR3)
subset1$d2 <- mahalanobis(subset1, colMeans(subset1), cov(subset1))
par(mfrow = c(1,2))
subset1$Z <- scale(subset1)
plot(subset1$d2, pch = 19)
plot(subset1$Z, pch = 19)
```

```{r}
subset2 <- subset1 %>% arrange(desc(d2), desc(Z)) 
# subset2[1,]
out <- which(subset1$L_CDR3 == 47)
# out
Data3 <- Data2[-972,]
```

### Response Variables  

We examined our responses: `H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq` and `Binding`. We observed that for `H_CDR3` the distributions were roughly normal with the center around 13 at different time-points (Figure \@ref(fig:hist-HCDR3T)) without taking into account different treatments. Figure \@ref(fig:hist-HCDR3) represents the distribution of `H_CDR3` with respect to treatments at different time-points, and slightly centered around 9 for `L_CDR3` at different time points. With `L_CDR3`, Figure \@ref(fig:hist-LCDR3T) and Figure \@ref(fig:hist-LCDR3) show approximately normal distribution with a longer right tail. The Q-Q plots in Figure \@ref(fig:QQ) show that `H_CDR3` and `L_CDR3` are both approximately normal.  

```{r, hist-HCDR3T, fig.cap = "Histogram H_CDR3"}
g3.1 <- ggplot(Data3, aes(H_CDR3))
  g3.1 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    xlab("HCDR3 Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    scale_x_continuous(breaks = round(seq(min(Data$H_CDR3), max(Data$H_CDR3), by = 1),1)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18))  #Title

```

```{r, hist-HCDR3, fig.cap = "Histograms of H_CDR3 vs Treatment and Timepoint"}
ggplot(data = Data3, mapping = aes(x = H_CDR3, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'H_CDR3', ylab = 'Count')
```


```{r, hist-LCDR3T, fig.cap = "Histogram L_CDR3"}
g5 <- ggplot(Data3, aes(L_CDR3))
  g5 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    xlab("LCDR3 Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    scale_x_continuous(breaks = round(seq(min(Data$L_CDR3), max(Data$L_CDR3), by = 1),1)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )
```


```{r, hist-LCDR3, fig.cap = "Histograms of L_CDR3 vs Treatment and Timepoint"}
ggplot(data = Data3, mapping = aes(x = L_CDR3, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'L_CDR3', ylab = 'Count')
```

```{r, QQ, fig.cap = "Q-Q Plots of H_CDR3 and L_CDR3"}
par(mfrow = c(1,2))
qqnorm(Data3$H_CDR3, pch = 19, cex = 1)
qqnorm(Data3$L_CDR3, pch = 19, cex = 1)
```

`HMuFreq` and `LMuFreq` were calculated by dividing `H_Substitution` by `H_VBase` for heavy chain and similarly for light chain. These two variables show how much the antibodies mutate. A higher mutation rate is usually indicative of better virus neutralization. Below we present comparison of mutation rate between heavy chain and light chain. Figure \@ref(fig:murate1), Figure \@ref(fig:murate2), Figure \@ref(fig:murate3), and Figure \@ref(fig:murate4) show that `HMuFreq` and `LMuFreq` are both approximately normally distributed, each with a longer right tail. The Q-Q plots in Figure \@ref(fig:QQ2) confirm the approximate normality of `HMuFreq` and `LMuFreq`.  

```{r, murate1, fig.cap = "Histogram HMuFreq"}
#par(mfrow = c(1,2))
g7 <- ggplot(Data3, aes(HMuFreq))
  g7 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    xlab("HMuFreq Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )
```

```{r, murate2, fig.cap = "Histograms of HMuFreq vs Treatment and Timepoint"}
ggplot(data = Data3, mapping = aes(x = HMuFreq, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'HMuFreq', ylab = 'n Monkeys')
```

```{r, murate3, fig.cap = "Histogram LMuFreq"}
g9 <- ggplot(Data3, aes(LMuFreq))
  g9 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    xlab("LMuFreq Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )
```

```{r, murate4, fig.cap = "Histograms of LMuFreq vs Treatment and Timepoint"}
ggplot(data = Data3, mapping = aes(x = LMuFreq, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'LMuFreq', ylab = 'n Monkeys')
```

```{r, QQ2, fig.cap = "Q-Q Plot of HMuFreq and LMuFreq"}
par(mfrow = c(1,2))
qqnorm(Data3$HMuFreq, pch = 19, cex = 1)
qqnorm(Data3$LMuFreq, pch = 19, cex = 1)
```

Next, a histogram of `Binding` with respect to treatments at different time points and Q-Q plot are shown in Figure \@ref(fig:hist-binding) and Figure \@ref(fig:QQ3). We observe that `Binding` was not normally distributed. However, since our sample size is larger than 2000, we can use the Central Limit Theorem and assume normality. Lastly, we check whether response variables could be correlated, as shown in Figure \@ref(fig:QQbinding). In these plots, we observe that none of the response variables were highly correlated.  

```{r, hist-binding, fig.cap = "Histograms of Binding Strength vs Treatment and Timepoint"}
ggplot(data = Data3, mapping = aes(x = Binding, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'Binding', ylab = 'Count')
```

```{r, QQ3, fig.cap = "Q-Q Plot of Binding"}
qqnorm(Data3$Binding, pch = 19, cex = 1)
```


```{r, QQbinding, fig.cap = "Plots of response variables"}
Data3 %>% select(LMuFreq, HMuFreq, L_CDR3, H_CDR3, Binding) %>% ggpairs()
```

## Multivariate Data Analysis  

To answer **Q1** (Do drugs and isotypes have effects on the mutation frequency and/or the amino acid count in the third complementarity determining region (CDR3)?), we test whether predictors `Drug` and `Isotype` had effects on the five responses: `H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq`, and `Binding`. 

### MANOVA

Since there are more than two populations we are comparing, we use MANOVA to test for effects. We check that the normality assumption is met due to large sample size (n = 2464). Each antibody is assumed to be independent. [equal variances?] 

Our formula for the MANOVA test is **(H_CDR3, HMuFreq, L_CDR3, LMuFreq, Binding)^T^ ~ Drug + Isotype.** In the output, both the main effects of `Drug` and `Isotype` have very low p-values, suggesting that both main effects are significant.  

```{r}
ID <- as.factor(Data3$MonkeyID)
drug <- as.factor(Data3$Drug)
tp <- as.factor(Data3$Time_Point)
it <- as.factor(Data3$Isotype)
fit.manova <- manova(cbind(Data3$H_CDR3, Data3$HMuFreq, Data3$L_CDR3, Data3$LMuFreq, Data3$Binding) ~ drug + it)
summary(fit.manova, test = "Wilks")
```

### Pairwise comparison

To know more details about which groups have different means, we use pairwise comparisons for each drug group and Isotype. (We set $\alpha$ as 0.05, and use a Bonferroni correction.)  

```{r}
respMat <- as.matrix(Data3[,c("L_CDR3", "LMuFreq", "H_CDR3", "HMuFreq", "Binding")])
# pairwise comparison among treatment groups
vars <- c("L_CDR3", "LMuFreq", "H_CDR3", "HMuFreq", "Binding")
```

```{r, pcdrug}
# pairwise comparison among drug groups
fit2 <- manova(respMat[,1:5] ~ drug)
# summary(fit2)
p <- 5
q2 <- length(unique(drug))
alpha.old <- 0.05
nc2 <- p*q2*(q2-1)/2
alpha.new2 <- alpha.old/nc2
for (i in 1:5){
  w <- c(0, 0, 0, 0, 0)
  w[i] <- 1
  print(paste(vars[i], " pairwise CI's"))
  cont <- contrast(emmeans(fit2, "drug", weights = w), "pairwise")
  bb <- confint(cont, level = 1 - alpha.new2, adj = "none")
  print(bb)
}
```

```{r}
# pairwise comparison among time point
fit3 <- manova(respMat[,1:5] ~ it)
# summary(fit3)
p <- 5
q3 <- length(unique(it))
alpha.old <- 0.05
nc3 <- p*q3*(q3-1)/2
alpha.new3 <- alpha.old/nc3
for (i in 1:5){
  w <- c(0, 0, 0, 0, 0)
  w[i] <- 1
  print(paste(vars[i], " pairwise CI's"))
  cont <- contrast(emmeans(fit3, "it", weights = w), "pairwise")
  bb <- confint(cont, level = 1 - alpha.new3, adj = "none")
  print(bb)
}
```

```{r, pctab}
col_drug <- t(data.frame(L_CDR3 = "none", LMuFreq = "none", H_CDR3 = "1 > 2", HMuFreq = "1 < 2, 1 > 3, 2 > 3", Binding = "1 < 3, 2 < 3"))
col_isotype <- t(data.frame(L_CDR3 = "none", LMuFreq = "none", H_CDR3 = "IgD < IgG", HMuFreq = "IgD < IgG, IgG > IgM", Binding = "IgA < IgG, IgD < IgG, IgG > IgM"))
pairTab <- cbind(col_drug, col_isotype)
colnames(pairTab) <- c("Drug", "Isotype")
kable(pairTab, caption = "Pair comparison of Isotype and Drug")
```

Table \@ref(tab:pctab) presents the pairs that have significant differences. We can see that `L_CDR3` and `LMuFreq` do not have significant paired differences for any drug groups and isotypes. For `H_CDR3`, drug 1 has a higher mean than drug 2, and IgG has a higher mean than IgD. For `HMuFreq`, drug 2 has the highest mean, followed by drug 1 and drug 3 (control), and IgG has higher mutation rate than IgD and IgM. For `Binding`, drug 3 (control) has the highest mean, but drug 1 and drug 2 do not have significant differences. IgG has higher binding rate than IgA, IgD, and IgM. [Kan, how to interpret pairwise comparisons?]  

The results show that the drug groups do increase mutation rate for heavy chain; however, they do not increase binding rate. Among all isotypes, IgG has the highest mutation rate for heavy chain and binding rate. We conclude that, although the treatments do help increase the diversity of antibodies, they are not specific to the HIV antigens and thus do not increase binding.  

## Longitudinal Analysis  

To answer our **Q2** (How does the binding strength of the antibodies develop in response to the number of vaccine dosages by treatment and drug types?), we use longitudinal data analysis, including general linear models and linear mixed models. For the longitudinal analysis of binding strength vs number of vaccine doses, we use the `gls` and `lme` functions from the nlme package$^{[7]}.$  

### One Covariate: Time Point  

```{r, mean-monkey, fig.cap = "Mean trend by monkey"}
ggplot(Data3, aes(x = Time_Point, y = Binding, colour = as.factor(MonkeyID))) + geom_point() + stat_summary(fun = mean, geom = "line")
```

```{r, variance-monkey, fig.cap = "Variances over time by monkey"}
ggplot(Data3, aes(x = Time_Point, y = Binding, colour = as.factor(MonkeyID))) + geom_point() + stat_summary(fun = var, geom = "line")
```

We first take a look at the data over time. As seen in Figure \@ref(fig:mean-monkey) and Figure \@ref(fig:variance-monkey), the mean trend is not linear, and the different time points have different variances. This information suggests that we should use piecewise linear models and set variances as unequal over time.  

We first consider a model with time point as the only covariate: $$Y_{ij}=\beta_0+\beta_1 Time_{ij}+e_{ij}$$  

We then turn the model above into a piecewise linear model, in which each segment has different intercepts and slopes. We use three indicator variables: $S1, S2, S3$ as the indicator variables, where 
$$S1=
\begin{cases}
1 & \text{if } 0 \le \text{Timepoint} < 1 \\
0 & \text{otherwise}
\end{cases}
$$
$$S2=
\begin{cases}
1 & \text{if } 1 \le \text{Timepoint} < 2 \\
0 & \text{otherwise}
\end{cases}
$$
$$S3=
\begin{cases}
1 & \text{if Timepoint} \ge 2 \\
0 & \text{otherwise}
\end{cases}
$$

The new model is thus $$Y_{ij}=S1(\beta_0+\beta_1 Time_{ij})+S2(\beta_2+\beta_3 Time_{ij})+S3(\beta_4+\beta_5 Time_{ij})+e_{ij}$$  

We also want to make sure that the trend is continuous at time points = 1 and 2. Our first complete model (`fit.gls`) is now $Y_{ij} = \beta_0(S1 + 2S2 - S2 Time_{ij}) + \beta_1 (S1 Time_{ij} + 2S2 - S2 Time_{ij}) + \beta_4(-S2 + S2 Time_{ij} + S3) + \beta_5(-2S2 + 2S2 Time_{ij} + S3 Time_{ij}) + e_{ij}$ where $$\mathbf{e}_i \sim N(0, \sigma^2I)$$  

```{r}
S1 <- as.numeric(Data3$Time_Point >= 0 & Data3$Time_Point < 1)
S2 <- as.numeric(Data3$Time_Point >= 1 & Data3$Time_Point < 2)
S3 <- as.numeric(Data3$Time_Point >= 2)
dataLDA1 <- data.frame(id = Data3$MonkeyID, binding = Data3$Binding, time = Data3$Time_Point, S1 = S1, S2 = S2, S3 = S3)
meanform <- binding ~ -1 + I(S1 + 2*S2 - S2:time) + I(S1:time + 2*S2 - S2:time) + I(-S2 + S2:time + S3) + I(-2*S2 + 2*S2:time + S3:time)
fit.gls <- gls(model = meanform, data = dataLDA1, weights = varIdent(form = ~ 1 | time), correlation = corCompSymm(form = ~1 | id))
# summary(fit.gls)
```

The model can also be written as $$Y_{ij} = S1(\beta_0) + S1 Time_{ij} (\beta_1) + S2( 2\beta_0 + 2\beta_1 - \beta_4 -2\beta_5) + S2 Time_{ij}(-\beta_0 -\beta_1 + \beta_4 + 2\beta_5)$$
$$ + S3(\beta_4) + S3 Time_{ij} (\beta_5) + e_{ij}$$  


```{r}
# when S1 = 1
f1 <- function(x, y){y = -0.2221651+0.2432183*x}
# when S2 = 1
f2 <- function(x, y){y = -0.2413024 + 0.2623556*x}
# when S3 = 1
f3 <- function(x, y){y = 0.7699600-0.2432756*x}
```

```{r, piecewise1, fig.cap = "Piecewise Linear Function--three segments"}
ggplot(Data3, aes(x = Time_Point, y = Binding, colour = as.factor(MonkeyID))) + 
  geom_point() + stat_function(fun = f1, xlim = c(0, 1), geom = "line") + 
  stat_function(fun = f2, xlim = c(1, 2), geom = "line") + 
  stat_function(fun = f3, xlim = c(2, 3), geom = "line")
```

From the model above, we can find the intercepts and slopes for all three segments of the mean trend:  

* S1: $-0.2221651+0.2432183*time$
* S2: $(2*-0.2221651+2*0.2432183-0.7699600+2*0.2432756)+(0.2221651-0.2432183+0.7699600-2*0.2432756)*time = -0.2413024 + 0.2623556*time$
* S3: $0.7699600-0.2432756*time$  

We make a plot of the line segments in Figure \@ref(fig:piecewise1), which shows the two segments S1 and S2 have very similar slopes. So we can refit the model with only two piecewise sections between time points 0 and 2 and between time points 2 and 3. We'll call them S4 and S5. The new model is therefore $$Y_{ij}=S4(\beta_0+\beta_1 Time_{ij})+S5(\beta_2+\beta_3 Time_{ij})+e_{ij}$$  

$$S4=
\begin{cases}
1 & \text{if Timepoint} < 2 \\
0 & \text{otherwise}
\end{cases}
$$
$$S5=
\begin{cases}
1 & \text{if Timepoint} \ge 2 \\
0 & \text{otherwise}
\end{cases}
$$

Again, we want to make sure that the trend is continuous at Time_Point = 2. Our second complete model (`fit.gls2`) is then $Y_{ij} = \beta_1(-2S4 + S4 Time_{ij}) + \beta_2(S4 + S5) + \beta_3(2S4 + S5 Time_{ij}) + e_{ij}$ where $$\mathbf{e}_i \sim N(0, \sigma^2I)$$  

```{r}
S4 <- as.numeric(Data3$Time_Point < 2)
S5 <- as.numeric(Data3$Time_Point >= 2)
dataLDA1$S4 <- S4
dataLDA1$S5 <- S5
meanform2 <- binding ~ -1 + I(-2*S4 + S4:time) + I(S4 + S5) + I(2*S4 + S5:time)
fit.gls2 <- gls(model = meanform2, data = dataLDA1, weights = varIdent(form = ~ 1 | time), correlation = corCompSymm(form = ~1 | id))
# summary(fit.gls2)
```

The model can also be written as $Y_{ij} = S4(-2 \beta_1 + \beta_2 + 2\beta_3) + S4 Time_{ij}(\beta_1) + S5(\beta_2) + S5 Time_{ij}(\beta_3) + e_{ij}$  

We first find the mean trend for S4 and S5:  

* S4: $(-2*0.5310975 + 0.5720853 + 2*-0.0000723) + 0.5310975*time = -0.4902543 + 0.5310975*time$
* S5: $0.5720853 - 0.0000723*time$  

We can make the plot again to see if the model is reasonable, as shown in Figure \@ref(fig:piecewise2). Indeed, there is a linear line between Time_Point 0 and 2 and one between Time_Point 2 and 3. The two lines are contiuous at Time_Point 2. A comparison of AIC And BIC of these two models, shown in Table \@ref(tab:AIC-tab1), indicates that the second model (`fit.gls2`) is indeed a better model.  

```{r}
# when S4 = 1
f4 <- function(x, y){y = -0.4902543 + 0.5310975*x}
# when S5 = 1
f5 <- function(x, y){y = 0.5720853 - 0.0000723*x}
```

```{r, piecewise2, fig.cap = "Piecewise Linear Function--two segments"}
ggplot(Data3, aes(x = Time_Point, y = Binding, colour = as.factor(MonkeyID))) + 
  geom_point() + stat_function(fun = f4, xlim = c(0, 2), geom = "line") + 
  stat_function(fun = f5, xlim = c(2, 3), geom = "line")
```

```{r, AIC-tab1}
knitr::kable(data.frame(cbind(AIC(fit.gls, fit.gls2), BIC(fit.gls, fit.gls2))), caption = "AIC and BIC between two gls models")
```

Next we check whether adding random effects improve our second complete model (`fit.gls2`). We assume that random effects exist in the intercept and slope. Our linear mixed model is then: $Y_{ij}= \beta_1(-2S4 + S4 Time_{ij}) + \beta_2(S4 + S5) + \beta_3(2S4 + S5 Time_{ij}) + b_{0i} + b{1i} Time_{ij} + e_{ij}$  
where $$\mathbf{b}_i \sim N \left( 0, \mathbf{D} = 
\begin{bmatrix}
D_{11} & D_{12} \\
 & D_{22}
 \end{bmatrix} \right)$$ and $$\mathbf{e}_i \sim N(0, \sigma^2I)$$  

```{r}
# with random intercept and slope, compound symmetry, unequal variances
fit.a1 <- lme(fixed = meanform2, random = ~ time | id, data = dataLDA1, weights = varIdent(form = ~ 1 | time), correlation = corCompSymm(form = ~1 | id))
# summary(fit.a1)
```

```{r}
# random intercept and slope, AR1, unequal variances
fit.a2 <- lme(fixed = meanform2, random = ~ time | id, data = dataLDA1, weights = varIdent(form = ~ 1 | time), correlation = corAR1(form = ~ 1 | id))
```

```{r, AIC-tab2}
kable(data.frame(AIC(fit.gls2, fit.a1, fit.a2), BIC(fit.gls2, fit.a1, fit.a2)), caption = "AIC and BIC for three models")
```

We fit two models with random effects: `fit.a1` assumes random intercept and slope for time point, compound symmetric correlation structure, and unequal variances over time; and `fit.a2` assumes random intercept and slope for time point, AR1 correlation structure, and uneuqal variances over time. As shown in Table \@ref(tab:AIC-tab2), the model `fit.a2` has the lowest AIC And BIC, so it seems the best model. We now check residuals for three models: `fit.gls2`, `fit.a1`, `fit.a2`, as shown in Figure \@ref(fig:QQmodel). All three Q-Q plots show approximate normality. To further investigate the effects of drugs, We now use `fit.a2` for further analysis.  

```{r, QQmodel, fig.cap = "Q-Q plots of models: GLS, compound symmetry, AR1"}
par(mfrow = c(1,3))
gls.res <- residuals(fit.gls2, type = "normalized")
qqnorm(gls.res)
lme.res1 <- residuals(fit.a1, level = 1, type = "normalized")
qqnorm(lme.res1)
lme.res2 <- residuals(fit.a2, level = 1, type = "normalized")
qqnorm(lme.res2)
```


Now we would like to know if the slopes between Time_Point 0 and 2 and between Time_Point 2 and 3 equal zero. $H_0: \text{ slope of } S4=0 \text{ and slope of } S5=0$, which means $H_0: \beta_1 = 0 \text{ and } \beta_3=0$  

Thus, we can check for two tests: $$\mathbf{L_1 \beta}=0$$ where $\mathbf{L_1}=(1, 0, 0)$ and $\mathbf{\beta}=(\beta_1, \beta_2, \beta_3)^T$ and $$\mathbf{L_2 \beta} = 0$$ where $\mathbf{L_2}=(0, 0, 1)$ and $\mathbf{\beta}=(\beta_1, \beta_2, \beta_3)^T$  

```{r, S4S5-slope}
L1 <- matrix(c(1, 0, 0), byrow = TRUE, nrow = 1)
L2 <- matrix(c(0, 0, 1), byrow = TRUE, nrow = 1)
test1 <- anova.lme(fit.a2, L = L1, adjustSigma = TRUE)
test2 <- anova.lme(fit.a2, L = L2, adjustSigma = TRUE)
kable(data.frame(rbind(test1, test2)), caption = "Inference of S4 ad S5 slopes")
```

As shown in Table \@ref(tab:S4S5-slope), the slop of S4 has a very small p-value, while the slope of S5 is quite large, indicating that the change in Binding rate between Time_Point 0 and Time_Point 2 is significant while the change between Time_Point 2 and Time_Point 3 is not significant. We conclude that Time_Point 2, when the monkeys had received two vaccines, had the highest Binding rate, while the last vaccine shot at Time_Point 3 did not make a difference to the Binding rate.  

### Two Covariates: Time Point and Drug  

Next we add `Drug` as a covariate to the model `gls.a2` to see if it has effects on Binding. We use two indicator variables: `D2` and `D3`, where  

$$D2=
\begin{cases}
1 & \text{if Drug = 2} \\
0 & \text{otherwise}
\end{cases}
$$

$$D3=
\begin{cases}
1 & \text{if Drug = 3} \\
0 & \text{otherwise}
\end{cases}
$$

Assuming that the random effects are the same for each drug, our model (`fit.a3`) with the extra covariate `Drug` is: $$Y_{ij}= \beta_1(-2S4 + S4 Time_{ij}) + \beta_2(S4 + S5) + \beta_3(2S4 + S5 Time_{ij}) +$$ 
$$\beta_4 D2 (-2S4 + S4 Time_{ij}) + \beta_5 D2 (S4 + S5) + \beta_6 D2(2S4 + S5 Time_{ij}) +$$ 
$$\beta_7 D3(-2S4 + S4 Time_{ij}) + \beta_8 D3(S4 + S5) + \beta_9 D3 (2S4 + S5 Time_{ij}) + b_{0i} + b{1i} Time_{ij} + e_{ij}$$  

where $$\mathbf{b}_i \sim N \left( 0, \mathbf{D} = 
\begin{bmatrix}
D_{11} & D_{12} \\
 & D_{22}
 \end{bmatrix} \right)$$ and $$\mathbf{e}_i \sim N(0, \sigma^2I)$$  
 
```{r}
dataLDA1$D2 <- as.numeric(Data3$Drug == 2)
dataLDA1$D3 <- as.numeric(Data3$Drug == 3)

v1 <- dataLDA1$S4 * -2  + dataLDA1$S4 * dataLDA1$time
v2 <- dataLDA1$S4 + dataLDA1$S5
v3 <- dataLDA1$S4 * 2 + dataLDA1$S5 * dataLDA1$time
v4 <- dataLDA1$S4 * -2 * dataLDA1$D2 + dataLDA1$S4 * dataLDA1$time * dataLDA1$D2
v5 <- dataLDA1$S4 * dataLDA1$D2 + dataLDA1$S5 * dataLDA1$D2
v6 <- dataLDA1$S4 * 2 * dataLDA1$D2 + dataLDA1$S5 * dataLDA1$time * dataLDA1$D2
v7 <- dataLDA1$S4 * -2 * dataLDA1$D3 + dataLDA1$S4 * dataLDA1$time * dataLDA1$D3
v8 <- dataLDA1$S4 * dataLDA1$D3 + dataLDA1$S5 * dataLDA1$D3
v9 <- dataLDA1$S4 * 2 * dataLDA1$D3 + dataLDA1$S5 * dataLDA1$time * dataLDA1$D3

meanform3 <- binding ~ -1 + v1 + v2 + v3 + v4 + v5 + v6 + v7 + v8 + v9

fit.a3 <- lme(fixed = meanform3, random = ~ time | id, data = dataLDA1, weights = varIdent(form = ~ 1 | time), correlation = corAR1(form = ~1 | id))

# summary(fit.a3)
```

With the model constructed, we want to make inference on $\boldsymbol{\beta}$ to find whether the drugs have any effects. To see whether Drug 1 and Drug 2 have any difference, we want to perform a hypothesis test on $H_0: \beta_4 = \beta_5 = \beta_6 = 0$, thus we can do the test $$\mathbf{L_3 \beta}=0$$ where $$\mathbf{L_3}=
\begin{bmatrix}
0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 \\
0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 \\
0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 
\end{bmatrix}$$ and $\mathbf{\beta}=(\beta_1, \beta_2, \beta_3, \beta_4, \beta_5, \beta_6, \beta_7, \beta_8, \beta_9)^T$  

To see whether Drug 1 and Drug 3 have any difference, we want to perform a hypothesis test on $H_0: \beta_7 = \beta_8 = \beta_9 = 0$, thus we can do the test $$\mathbf{L_4 \beta}=0$$ where $$\mathbf{L_4}=
\begin{bmatrix}
0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 \\
0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 \\
0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 
\end{bmatrix}$$ and $\mathbf{\beta}=(\beta_1, \beta_2, \beta_3, \beta_4, \beta_5, \beta_6, \beta_7, \beta_8, \beta_9)^T$  

To see whether Drug 2 and Drug 3 have any difference, we want to perform a hypothesis test on $H_0: \beta_4 = \beta_7, \beta_5 = \beta_8, \beta_6 = \beta_9$, thus we can do the test $$\mathbf{L_5 \beta}=0$$ where $$\mathbf{L_5}=
\begin{bmatrix}
0 & 0 & 0 & 1 & 0 & 0 & -1 & 0 & 0 \\
0 & 0 & 0 & 0 & 1 & 0 & 0 & -1 & 0 \\
0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & -1 
\end{bmatrix}$$ and $\mathbf{\beta}=(\beta_1, \beta_2, \beta_3, \beta_4, \beta_5, \beta_6, \beta_7, \beta_8, \beta_9)^T$  

We found that, as shown in Table \@ref(tab:L3-test), Drug 1 and Drug 2 do not have significantly different effects on Binding rates. As shown in Table \@ref(tab:L4-test), Drug 1 and Drug 3 do not have significantly different effects on Binding rates. Also, as shown in Table \@ref(tab:L5-test), Drug 2 and Drug 3 do not have significantly different effects on Binding rates. In other words, drug groups do not have signifant effects on our longitudinal model. Thus we will retain `fit.a2` as our best model.  

```{r}
betahat <- fit.a3$coefficients$fixed
V.robust <- vcovCR(fit.a3, type = "CR0")
df <- nrow(dataLDA1) - length(betahat)
```

```{r, L3-test}
# F-test
L3 <- matrix(c(0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0), byrow = TRUE, nrow = 3)
varmat3 <- L3 %*% V.robust %*% t(L3)
c3 <- nrow(L3)
est_L3 <- L3 %*% betahat
Fstat_L3 <- c( t(est_L3) %*% solve(varmat3) %*% est_L3 ) / c3
p.value <- pf(q = Fstat_L3, df1 = c3, df2 = df, lower.tail = FALSE)
knitr::kable(data.frame(Fstat = Fstat_L3, p_value = p.value), caption = "Test whether drug 1 = drug 2")
```

```{r, L4-test}
# F-test
L4 <- matrix(c(0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1), byrow = TRUE, nrow = 3)
varmat4 <- L4 %*% V.robust %*% t(L4)
c4 <- nrow(L4)
est_L4 <- L4 %*% betahat
Fstat_L4 <- c( t(est_L4) %*% solve(varmat4) %*% est_L4 ) / c4
p.value <- pf(q = Fstat_L4, df1 = c4, df2 = df, lower.tail = FALSE)
knitr::kable(data.frame(Fstat = Fstat_L4, p_value = p.value), caption = "Test whether drug 1 = drug 3")
```

```{r, L5-test}
# F-test
L5 <- matrix(c(0, 0, 0, 1, 0, 0, -1, 0, 0, 0, 0, 0, 0, 1, 0, 0, -1, 0, 0, 0, 0, 0, 0, 1, 0, 0, -1), byrow = TRUE, nrow = 3)
varmat5 <- L5 %*% V.robust %*% t(L5)
c5 <- nrow(L5)
est_L5 <- L5 %*% betahat
Fstat_L5 <- c( t(est_L5) %*% solve(varmat5) %*% est_L5 ) / c5
p.value <- pf(q = Fstat_L5, df1 = c5, df2 = df, lower.tail = FALSE)
knitr::kable(data.frame(Fstat = Fstat_L5, p_value = p.value), caption = "Test whether drug 2 = drug 3")
```

# Results  

For multivariate analyses, we performed a MANOVA test on the main effects of `Isotype` and `Drug` on the response variable vector (`H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq`, `Binding`)^T^. We found that both `Isotype` and `Drug` have very small p-values. We also performed pairwise comparison to see where the effects were.  

For longitudinal analyses, we found that the best linear mixed model is $Y_{ij}= \beta_1(-2S4 + S4 Time_{ij}) + \beta_2(S4 + S5) + \beta_3(2S4 + S5 Time_{ij}) + b_{0i} + b{1i} Time_{ij} + e_{ij}$  
where $$\mathbf{b}_i \sim N \left( 0, \mathbf{D} = 
\begin{bmatrix}
D_{11} & D_{12} \\
 & D_{22}
 \end{bmatrix} \right)$$ and $$\mathbf{e}_i \sim N(0, \sigma^2I)$$  
 
Using inferences on $\boldsymbol{\beta}$ for the two line segments, We rejected the hypothesis that the slope of the first line segment (between time points 0 and 2) is zero, but we failed to reject the hypothesis that the slope of hte second line segment (between time points 2 and 3) is zero. We then added `Drug` as another covariate to the above model and made inference on $\boldsymbol{\beta}$. The comparison between three drug groups was done with three F-tests (between Drug grups 1 and 2, 2 and 3, and 1 and 3), which led to three p-values greater than 0.05. Thus We failed to reject the null hypotheses that the mean trend of either of the two drug groups was equal.  

# Discussion  

While our analyses reached some findings, some further investigations could improve our analyses. A common method to analyze antibody traits is to treat antibodies (rows in our data) as independent from each other. We considered perform statistical analyses to prove or disprove such a convention, but it requires more biological knowledge and the investigation would most likely be beyond the scope of a final report. However, this remains an interesting topic that could be explored.  

For longitudinal data analyses, we did not try out more combinations for models. For example, we only tried two correlation structures (compound symmetry and AR1); other structures might achieve better results. When we added drug as another covariate, we did not go back to test which correlation structure might perform better or whether the piecewise model should include two or three line segments.  

# Conclusions  

In this report, we performed multivariate data analyses and longitudinal data analyses to understand whether time points, drugs, and isotypes have effects on characteristics of antibodies. Our statistical analyses provide answers to our two research questions. We performed a MANOVA test to our first research question, "Do drugs and isotypes have effects on the mutation frequency and/or the amino acid count in the third complementarity determining region (CDR3)?" and found significant main effects for both drugs and isotypes. However, pairwise comparisons reveal that drugs increased mutation rates for heavy chain but did not increase binding rates, which are the key to better efficacy of vaccines. [Kan: how to interpret IgG's higher mutation rates and binding rates?]
  
To answer our second research question, How does the binding strength of the antibodies develop in response to the number of vaccine dosages by treatment and drug types?", we first used time point as the only covariate and constructed with two general linear models with two and three line segments. We then added random effects for intercept and slope for time point as well as different correlation structure--compound symmetry and AR1--and found the model with two line segments (time points 0 to 2 and time points 2 to 3), random effects of intercept and slope of time point, AR1 correlation structure, and unequal variances over time performs best. F-tests for inferences for $\boldsymbol{\beta}$ reveals that time point 2 have the highest binding rates, suggesting that two vaccine injections improved the binding rates while the third injection did not make a difference. We also found that adding drug as a covariate did not improve the model.  

# References  

The dataset, [which can be found here](https://github.com/luokan1227/537P1), was provided by Kan Luo, as he was one of authors for the following four publications that used the dataset:  

1. Luo K, Liao HX, Zhang R, et al. Tissue memory B cell repertoire analysis after ALVAC/AIDSVAX B/E gp120 immunization of rhesus macaques. *JCI Insight*. 2016;1(20):e88522. Published 2016 Dec 8. doi:10.1172/jci.insight.88522
2. Bradley, T., Kuraoka, M., Yeh, C.-H., Tian, M., Chen, H., Cain, D. W., . . . Haynes, B. F. (2020). Immune checkpoint modulation enhances HIV-1 antibody induction. *Nature Communications*, 11(1), 948. doi:10.1038/s41467-020-14670-w
3. Easterhoff, D., Pollara, J., Luo, K., Tolbert, W. D., Young, B., Mielke, D., . . . Ferrari, G. (2020). Boosting with AIDSVAX B/E Enhances Env Constant Region 1 and 2 Antibody-Dependent Cellular Cytotoxicity Breadth and Potency. *Journal of Virology*, 94(4), e01120-01119. doi:10.1128/jvi.01120-19
4. Wiehe, K., Easterhoff, D., Luo, K., Nicely, N. I., Bradley, T., Jaeger, F. H., Dennison, S. M., Zhang, R., Lloyd, K. E., Stolarchuk, C., Parks, R., Sutherland, L. L., Scearce, R. M., Morris, L., Kaewkungwal, J., Nitayaphan, S., Pitisuttithum, P., Rerks-Ngarm, S., Sinangil, F., Phogat, S., . Haynes, B. F. (2014). Antibody light-chain-restricted recognition of the site of immune pressure in the RV144 HIV-1 vaccine trial is phylogenetically conserved. *Immunity*, 41(6), 909-918. https://doi.org/10.1016/j.immuni.2014.11.014  
5. Lefranc MP, Giudicelli V, Ginestoux C, Bodmer J, Muller W, Bontrop R, Lemaitre M, Malik A, Barbie V, Chaume D. IMGT, the international ImMunoGeneTics database. *Nucleic Acids Res*. 1999;27:209-212. doi: 10.1093/nar/27.1.209.  
6. Jenny M Woof , Dennis R Burton,Human antibody-Fc receptor interactions illuminated by crystal structures.*Nat Rev Immunol*. 2004 Feb;4(2):89-99. doi: 10.1038/nri1266.  
7. Pinheiro J, Bates D, DebRoy S, Sarkar D, R Core Team (2020). _nlme: Linear and Nonlinear Mixed Effects Models_. R package
