---
title: "Final Project -- **Antibody Response Induced by HIV Vaccines and T-cell Suppression Treatments in Rhesus Macaques** -- Second Draft"
author: "Kan Luo, Shih-Ni Prim, Frederick Davey, Rizwana Rehman"
date: "10/25/2020"
output: 
  pdf_document:
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, eval = TRUE, message = FALSE, warning = FALSE)
library(readxl)
library(tidyverse)
library(ggplot2)
library(httr)
library(nlme)
library(emmeans)
library(GGally)
library(knitr)
```

# Notes/questions  

* The resulting dataset from this section is called `Data2`, and one outlier will be removed in a later section, which results in a final dataset `Data3`. So `Data3` is used for analysis.
* Questions for Kan marked as [Kan...] below.
* Comments for future addition and revisions in [...]
* Instead of looking at average of Binding, look at average of reactivity to see the percentage of reactive. Perhaps use natural log to transform Binding.
* Who can figure out how to put all figures and tables after all text and before supplmental materials?
* Dig deeper into analysis results?
* Not sure what sections the professor wants  

```{r readin Data, results = 'hide'}
#Readin data from Github url address
url1 <- 'https://github.com/luokan1227/537P1/raw/master/Data.xlsx'  # main data
url2 <- 'https://github.com/luokan1227/537P1/raw/master/MonkeyID.xlsx'  # Monkey ID
GET(url1, write_disk(tf1 <- tempfile(fileext = ".xlsx")))
GET(url2, write_disk(tf2 <- tempfile(fileext = ".xlsx")))
Rawdata <- read_excel(tf1)
MonkeyID <-  read_excel(tf2)
#Add Monkey ID to raw data
Rawdata <- inner_join(Rawdata, MonkeyID, by = "H_ID")
colnames(Rawdata)[2] <- "Time_Point"
#-----------------------
#Adjust or add variables
#-----------------------

# Final data set for all following analysis
Data <- Rawdata

# add drug type and reactivity
Data$Drug <- ifelse(Data$Treatment == "group 7", 3, 
                      ifelse(Data$Treatment == "group 1", 1, 
                      ifelse(Data$Treatment == "group 2", 1, 
                      ifelse(Data$Treatment == "group 3", 1, 2))))

Data$Reactivity <- ifelse(Data$Binding > 0.1, 1, 0)

# create a subset with the variables we need plus the extracted information
Data2 <- Data %>% select(MonkeyID, Drug, Treatment, Time_Point, Isotype, H_VBase, H_Substitutions, H_Insertions, H_Deletions, HMuFreq, H_CDR3, L_VBase, L_Substitutions, L_Insertions, L_Deletions, LMuFreq, L_CDR3, Binding, Reactivity)
```

# Abstract  

# Introduction  

A dominant vaccine development strategy is to induce neutralizing antibodies by immunizing humans with the virusâ€™ glycoproteins. However, HIV vaccines that adopted this strategy mostly failed due to the fact that HIV is an RNA virus, which mutates rapidly to escape the inhibition of neutralizing antibodies. By the time the body generates neutralizing antibodies against the glycoproteins of some HIV strains, the RNA virus has already mutated. Thus, the existing neutralizing antibody fails to recognize, bind with, and neutralize the HIV virus. One possible solution is to increase the number of potential neutralizing antibodies that will cycle in the body by releasing a variety of antibodies after glycoprotein immunization.  

Our dataset includes measurements of antibodies measured in 20 rhesus macaques after they were given the same HIV vaccine at three different time points and one of three randomly selected anti-Treg treatments. Blood samples were collected two weeks after vaccine dosing, and antibodies were isolated from those samples. A different number of antibodies were collected from each blood sample, limited by assay yield. Each observation contains information about the antibody isolated post the glycoprotein immunization. In the current report, we test if the different Treg treatments and number of vaccine injections cause changes in the antibody characteristics and if the changes are related to immunization/treatment timepoints.  

Specifically, we evaluate:  

1. **Do treatments, time points, or isotypes have effects on the mutation frequency or the amino acid count in the third complementarity determining region (CDR3)?** This will be evaluated with a three-way MANOVA, followed by further pair-wise analysis of any values found significantly different.  
  
2. **Do the paired antibody heavy chain and light chain have different mutation rate, CDR3, or binding values?** This will be evaluated with paired comparisons by antibody.  
  
3. **How does the binding strength of the antibodies develop in response to the number of vaccine dosages by treatment?** This will be evaluated with a longitudinal analysis by test subject.  

# Methodologies  

During the experiment, 20 rhesus macaques were given glycoprotein immunization and supplemental antibody doses, as well as one of three treatments (two experimental regulatory T-cell suppression treatments and one control). Regulatory T (Treg) cells prevent autoimmune diseases and suppress allergic reactions by inhibiting adaptive antibody immune response in the germinal center. Theoretically, this adaptive response lowers the effectiveness of vaccines. Thus the experiment used T-cell suppression treatments to investigate  the effect on immunization. These drugs are widely used in post transplant immunosuppression treatment to prevent rejection.  

For the analysis of mutation frequency and CDR3 count, each antibody within the same treatment is treated as an independent observation. While we might expect different variance within subject vs between subjects, the number of potential antibodies observed is much higher than the number sampled in a blood draw.  A human antibody is formed by a heavy chain and light chain. For heavy chain, human has about 51 V-gene segments, 25 D-gene segments and 6 J-gene segment. For light chain (kappa and lambda), there are 71 V-gene segments and 9 J-gene segments[ref.5]. Any heavy chain V-D-J combination and light chain V-J combinations can randomly happen in germline center. Theoretically, there can be $51*25*6*71*9=$ `r 51*25*6*71*9` combinations of gene segments. Considering the frequently happened mutation, insertion, class switching, each individual can have over **10 billion** different antibodies. Thus, we decided to follow the convention of vaccine studies and treat each antibody as independent.   [Kan -- can you identify a reference article or journal here that uses this convention? We don't need to quote it / change the answer, it's just defending our claim that it's a standard practice]

To evaluate the mutation frequency and CDR count vs Treatments, Timepoints, and Isotypes MANOVA, we will be running the [xxx] function from the [xxx] package [package citation]

For the longitudinal analysis of binding strength vs number of vaccine doses, we will be using the gls function from the nlme package$^{[7]}.$ 

# Data Summaries  

The dataset has `r nrow(Data2)` antibodies collected from 20 rhesus monkeys. We first present our exploratory data analysis and summaries.  First an overview of the number of antibodies collected per treatment at each timepoint

```{r, echo= FALSE}
#table(Data2$MonkeyID)
Data2Summary<- Data2 %>% group_by(Drug, Time_Point, MonkeyID) %>% summarize(nObs = n())
Data2Summary$Drug<- as.factor(Data2Summary$Drug)

g<- ggplot(data = Data2Summary, mapping = aes(x = nObs, color = Drug)) + 
  geom_histogram(bins = 8, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'n Antibodies Collected', ylab = 'n Monkeys', title= 'Histograms of Antibodies Collected per Monkey vs Treatment and Timepoint')
g

table(Data2$Drug, Data2$Time_Point, dnn= c('Drug', 'TimePoint'))
table(Data2$Drug, Data2$Time_Point, Data2$Isotype, dnn= c('Drug', 'TimePoint', 'Isotype'))

```
There are four time points; time zero was collected before any procedure was done. Times 1,2, and 3 were collected two weeks after an initial and two booster vaccine shots were administered to the macaques. In the treatment groups, groups 1-3 represent different doses of anti T-Reg drug 1, groups 4-6 represent different doses of anti-Treg drug 2, and group 7 represents the control group [We have 20 monkeys, do we need to mention groups at all?  Can we just stick to drugs 1,2, and control group 3?].  

Next, we'll take a look at the variable `Isotype`. There are 5 kinds of immunoglobulin isotypes: IgG, IgA, IgM, IgE, IgD *[ref.6]*. The two most important kinds are IgG and IgM. IgM occurs in the acute stage of infection and perform an role of primary response. The secondary response IgG appears later in serum with higher binding affinity, and neutralizing potentials against toxins and virus. IgA mostly found in mucosal tissues such as Nasal mucosa. Non-dominant IgD and IgE are typically lower than 1% in blood.  

```{r Ig Isotype Plot/Table}
# Histogram 
Iso.order <- c("G", "M", "A", "D", "E") 
g1 <- ggplot(Data, aes(x=factor(Isotype, levels = Iso.order), fill=as.factor(Drug))) 
  g1 + geom_bar(stat = "count", width = .9, position = 'Stack') + # his plot
    facet_grid(cols = vars(Time_Point), labeller = label_both) +
    labs(fill = 'Drug') + # Legend Content
    ggtitle("Ig Isotype") +  #Plot title
    xlab("Ig Isotype") + ylab("Number of abs") +# X axis and y axis title
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 20) #Title
          #legend.text = element_text(size=20), #Legend text
          #legend.title = element_text(size = 20) # Legend title
    )
```

As expected, we see that IgG and IgM occupy the biggest proportion of all antibodies in all time points. Before immunization (time point 0), there are similar weight of IgG and IgM found in blood. After the 1st immunization (time point 1), primary immune response results an increase of IgM, followed with IgG increase at later time point 2 and 3.  We'll use the variable `Isotype` as a grouping covariate later.  

For the response variables, we will begin with an outlier check, then review their distributions across treatment and timepoints.

```{r Outlier Review}
tibContinuous <- gather(Data2, key = "Variable", value = "value", 
                        H_CDR3, HMuFreq, L_CDR3, LMuFreq, Binding, Reactivity )
#histograms of continuous variables
ggplot(data = tibContinuous, aes(x = value)) + 
  geom_histogram(bins = 20) +
  facet_wrap(~Variable, scales = 'free') + labs(title = 'Histogram of Response Variables')

t<- summary(scale(Data2$L_CDR3))
kable(t(t), caption = 'Normal Standardized L_CDR3 Statistics')
```
L_CDR3 appears to have an outlier several standard.  This is most likely caused by a laboratory error, so we will exclude it and continue the analysis with the remaining data.

```{r}
Data3 <- Data2[-972,]
```

Next we'll examine our variables: `H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq`, `Binding`, and `Reactivity`. In each antibody, there are two sets of heavy chain and light chain, all of which forming a Y-shape immunoglobulin. Thus many of the variables start with H or L, indicating which chain the information comes from. 

`H_CDR3` and `L_CDR3` indicates the length of the third complementarity-determining region on the variable heavy chain and light chain. Post antigen stimulation, antibody obtains higher antigen-binding affinity by random mutations and insertions. For HIV antibody response, researchers found many broadly neutralizing antibodies, such as PG9, shared the characteristics of long HCDR3 and high mutation frequency *[ref.7]*. In HIV vaccine study, a goal is to induce more long HCDR3 and highly mutated antibodies, which are more potential to be HIV neutralizing antibody.  

Below we see that the distributions are roughly normal with the center around 13 amino acid for `H_CDR3`, with all data points, and slightly centers for different time points.

```{r CDR3 Plot/Table, echo=FALSE}
g3.1 <- ggplot(Data3, aes(H_CDR3))
  g3.1 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    ggtitle("H_CDR3") +  #Plot title
    xlab("HCDR3 Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    scale_x_continuous(breaks = round(seq(min(Data$H_CDR3), max(Data$H_CDR3), by = 1),1)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )

g4 <- ggplot(Data3, aes(L_CDR3))
  g4 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    ggtitle("L_CDR3") +  #Plot title
    xlab("LCDR3 Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    scale_x_continuous(breaks = round(seq(min(Data$L_CDR3), max(Data$L_CDR3), by = 1),1)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )

par(mfrow = c(1,2))
qqnorm(Data3$H_CDR3, main = "Q-Q Plot of H_CDR3", pch = 19, cex = 1)
qqnorm(Data3$L_CDR3, main = "Q-Q Plot of L_CDR3", pch = 19, cex = 1)
```
`HMuFreq` and `LMuFreq` are calculated by dividing `H_Substitution` by `H_VBase` for heavy chain and similarly for light chain. These two variables show how much the antibodies mutate. Their values are obtained by comparing our obtained antibody sequence to it's germline gene segement sequence. For the purpose of the study, the higher the mutation rate is, the better. Below we see some comparison of mutation rate between heavy chain and light chain. (Kappa and Lambda are two kinds of light chain.) [Kan, is there a reason to split up light chain into Kappa and Lambda? Could we simply plot heavy chain vs. light chain? *Yes. If you just want to talk about heavy chain, you can ignore the light chain is kappa or lambda. But if you want to discuss something on light chain, kappa and lambda light chains are different two subtypes, almost everyone in the area split them.*]  

```{r mutation rate, echo=FALSE, warning=FALSE}
#par(mfrow = c(1,2))
g5<- ggplot(data = Data3, mapping = aes(x = HMuFreq, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'HMuFreq', ylab = 'n Monkeys', title= 'Histograms of HMuFreq vs Treatment and Timepoint')
g5

g6<- ggplot(data = Data3, mapping = aes(x = LMuFreq, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'LMuFreq', ylab = 'n Monkeys', title= 'Histograms of LMuFreq vs Treatment and Timepoint')
g6

qqnorm(Data3$HMuFreq, main = "Q-Q Plot of HMuFreq", pch = 19, cex = 1)
qqnorm(Data3$LMuFreq, main = "Q-Q Plot of LMuFreq", pch = 19, cex = 1)
```


Lastly, `Binding` is mostly related with the ability of neutralizing, only those antibodies whom bind with the virus can have the chance to make the virus ineffective. This is the most important measure of the study. `Reactivity` turns `Binding` into a binary variable; `Binding` affinity above 0.1 is considered reactive.  

In the Q-Q plot of `Binding`, we can see that it is not normally distributed, because in theory the `Binding` affinity can't below 0, which indicates no binding at all. The upper boundary of testing `Binding` affinity is around 4 because of saturation of assay. Thus the `Reactivity` measure might be a better response to use. However, since our sample size is larger than 2000, we can use the Central Limit Theorem and assume normality.  

```{r}
g8<- ggplot(data = Data3, mapping = aes(x = Binding, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'Binding', ylab = 'Count', title= 'Histograms of Binding Strength vs Treatment and Timepoint')
g8

qqnorm(Data3$Binding)
```
Let's take a look at these variables and check whether they might be correlated. In the plot below, we can see that none of the variables are highly correlated, except for `Binding` and `Reactivity`, which is expected, because `Reactivity` is a binary variable derived from `Binding`. We will only choose one of these two response variables in each analysis based on the type of analysis. [What else to point out here?]  

```{r}
Data2 %>% select(LMuFreq, HMuFreq, L_CDR3, H_CDR3, Binding, Reactivity) %>% ggpairs()
```
* [We should trim this, move to the data analysis section if desired]
Now we use some plots to see whether the response variables might be different for different time points, treatment groups or drugs, and grouping covariate (`Isotype`).  

```{r}
ggplot(Data2, aes(x = Reactivity)) + geom_bar(aes(fill = as.factor(Time_Point)), position = "dodge") + facet_wrap( ~ Drug, labeller = label_both) + scale_fill_discrete(name = "Time Point")
ggplot(Data2, aes(x = Time_Point, y = Binding)) + geom_point() + facet_wrap( ~ Drug, labeller = label_both) + scale_fill_discrete(name = "Time Point")
table(Data$Drug, Data$Reactivity)
table(Data$Time_Point, Data$Reactivity)
```

`Binding` or `Reactivity` do seem to be affected by various predictors, and the boxplots for `Binding` do appear quite different.  

From the table Drug vs. Reactivity, we find among all three treatments, HIV-envelope negative antibodies are more than positive antibodies. However, the positive antibody percentage is higher in control group in general. This can be explained by T-reg cell inhibition treatments in group 1 and 2 lead an massive antibody release from germinal center, non-HIV-specific antibodies able to circling in blood without the suppression from T-reg cells, so the percentage of HIV-negative antibody become higher than normal group (group 3).  
From the table Time_Point vs. Reactivity, we can see there are very few HIV-positive antibodies in the base line (Time point 0). Those antibodies suppose to major bind with some other (unknown) antigens and happens to cross-reactive with HIV-envelopes, so their binding affinity to HIV envelopes are typically low. We can observe that from Time_Point vs. Binding plot above. As the time goes on and more vaccince immunization received, we are able to find HIV-positive antibodies in rhesus blood with significant increased rates.  


```{r}
Data2 %>% group_by(Drug) %>% summarize(avgLMuFreq = mean(LMuFreq), avgHMuFreq = mean(HMuFreq), avgBinding = mean(Binding), varBinding = var(Binding), avgReact = mean(Reactivity))
ggplot(Data2, aes(x = Drug, y = Binding)) + geom_point()

```

# Data Analysis  

## Multivariate Data Analysis  

Now we want to test whether predictors `Drug` and `Isotype` have effects on the five responses: `H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq`, and `Binding`. We choose `Binding` here, because all the variables are continuous. 

# [I don't think this is an accurate sample size.  Some of the isotypes only have 1-10 samples in some of the time points.  I think we should trim to IgG and IgM, can possibly include IgD]

First, we use manova to test effects. Since we have a large sample size (n = 2464), we can assume normality. In the output, we can see that the main effects of `Drug` and `Isotype` and the interaction effects are all significant.  

#[How are we managing controls? / Controlling for time 0 data?]

```{r}
ID <- as.factor(Data3$MonkeyID)
trt <- as.factor(Data3$Treatment)
drug <- as.factor(Data3$Drug)
tp <- as.factor(Data3$Time_Point)
it <- as.factor(Data3$Isotype)

fit.manova <- manova(cbind(Data3$H_CDR3, Data3$HMuFreq, Data3$L_CDR3, Data3$LMuFreq, Data3$Binding) ~ drug*it)
summary(fit.manova)
```
### Pairwise comparison

To know more details about which groups have different means, we use pairwise comparisons for each treatment group, drug, and Isotype. (We set $\alpha$ as 0.05, and use a Bonferroni correction.)  

```{r}
respMat <- as.matrix(Data3[,c("L_CDR3", "LMuFreq", "H_CDR3", "HMuFreq", "Binding")])
# pairwise comparison among treatment groups

vars <- c("L_CDR3", "LMuFreq", "H_CDR3", "HMuFreq", "Binding")

```

```{r}
# pairwise comparison among drug groups
fit2 <- manova(respMat[,1:5] ~ drug)
# summary(fit2)
p <- 5
q2 <- length(unique(drug))
alpha.old <- 0.05
nc2 <- p*q2*(q2-1)/2
alpha.new2 <- alpha.old/nc2

for (i in 1:5){
  w <- c(0, 0, 0, 0, 0)
  w[i] <- 1
  print(paste(vars[i], " pairwise CI's"))
  cont <- contrast(emmeans(fit2, "drug", weights = w), "pairwise")
  bb <- confint(cont, level = 1 - alpha.new2, adj = "none")
  print(bb)
}
```

```{r}
# pairwise comparison among time point
fit3 <- manova(respMat[,1:5] ~ it)
# summary(fit3)
p <- 5
q3 <- length(unique(it))
alpha.old <- 0.05
nc3 <- p*q3*(q3-1)/2
alpha.new3 <- alpha.old/nc3

for (i in 1:5){
  w <- c(0, 0, 0, 0, 0)
  w[i] <- 1
  print(paste(vars[i], " pairwise CI's"))
  cont <- contrast(emmeans(fit3, "it", weights = w), "pairwise")
  bb <- confint(cont, level = 1 - alpha.new3, adj = "none")
  print(bb)
}
```
# [ We should have code build these tables, why / where are we fitting for groups? This muddies the conclusions]
Here are the pairs that have significant differences:  

* Treatment
    * L_CDR3: none
    * LMuFreq: none
    * H_CDR3: 
        * group 1 > group 4
    * HMuFreq: 
        * group 1 > group 7
        * group 3 < group 5
        * group 3 < group 6
        * group 5 > group 7
        * group 6 > group 7
    * Binding:
        * group 1 < group 2
        * group 1 > group 6
        * group 2 > group 3
        * group 2 > group 4
        * group 2 > group 5
        * group 2 > group 6
        * group 3 < group 7
        * group 4 > group 6
        * group 4 < group 7
        * group 5 < group 7
        * group 6 < group 7
* Drug
    * L_CDR3: none
    * LMuFreq: none
    * H_CDR3: 
        * 1 > 2
    * HMuFreq: 
        * 1 < 2
        * 1 > 3
        * 2 > 3
    * Binding: 
        * 1 < 3
        * 2 < 3
* Isotype
    * L_CDR3: none
    * LMuFreq: none
    * H_CDR3: 
        * IgD < IgG
    * HMuFreq: 
        * IgD < IgG
        * IgG > IgM
    * Binding: 
        * IgA < IgG
        * IgD < IgG
        * IgG > IgM

In short, `L_CDR3` and `LMuFreq` do not have significnat paired differences.  

For `H_CDR3`, treatment group 1 (drug 1) is higher than treatment group 7 (control), and IgG has a longer H_CDR3 length than IgD.  

For `HMuFreq`, drug 2 has the highest mean, followed by drug 1 and control. More specifically, treatment groups 5 and 6 (two doses in drug 2) have the highest mutation rates. IgG has higher mutation rate than IgD.  

For `Binding`, drug 3 has the highest mean, but drug 1 and drug 2 do not have significant differences. IgG has higher binding rate than IgA, IgD, and IgM.  

We could conclude that the drugs/treatment groups do increase mutation rate; however, they do not increase binding rate. That is to say, although the treatments do help increase the diversity of antibodies, they are not specific to the HIV antigens and thus do not increase binding. 

## Longitudinal Data Analysis  
[This section probably needs some more revisions, since we still have three more lectures.]

First we don't consider treatments but only plot the mean trend over time. The plot shows that binding does vary over time. The red line shows the mean trend over time, and the blue line shows the variance over time. The variance does not seem equal over time, so we use unequal variance over time for the covariance structure.  

$$Y_{ij}=\beta_0+\beta_1 Time_{ij}+e_{ij}$$

```{r}
# simply connects the mean of each time point
ggplot(Data3, aes(x = Time_Point, y = Binding)) + geom_point(stat = "identity", aes(color = as.factor(Drug))) + geom_jitter() + stat_summary(fun = mean, geom = "line", lwd = 2, color = "red") + stat_summary(fun = var, geom = "line", lwd = 2, color = "blue") + scale_color_discrete(name = "Drug")

```

Here we use `Binding` as the response, `Time_Point` as the time factor, and `Drug` as the covariates. Random effect for both intercept and slope. Now we want to add one covariate: `Drug`. We use two indicator variables: `D1` and `D2`, where  

$$D1=
\begin{cases}
1 & \text{if Drug = 1} \\
0 & \text{otherwise}
\end{cases}
$$

$$D2=
\begin{cases}
1 & \text{if Drug = 2} \\
0 & \text{otherwise}
\end{cases}
$$

Assuming that the random effects are the same for each drug, our full model is: 

$$Y_{ij} = \beta_0+ \beta_1 Time_{ij} + D1_i(\beta_2+\beta_3 Time_{ij}) + D2_i (\beta_4+\beta_5 Time_{ij}) + b_{0i} + b_{1i} Time_{ij} + e_{ij}$$  
$$\mathbf{b_i} \sim N \left(0, \mathbf{D}=
\begin{bmatrix}
D_{11} & D_{12} \\
D_{12} & D_{22} 
\end{bmatrix}
\right)$$

Drug 1: $Y_{ij} = \beta_0+\beta_1 Time_{ij} + \beta_2+\beta_3 Time_{ij} + b_{0i} + b_{1i} Time_{ij} + e_{ij}$  
Drug 2: $Y_{ij} = \beta_0+\beta_1 Time_{ij} + \beta_4+\beta_5 Time_{ij} + b_{0i} + b_{1i} Time_{ij} + e_{ij}$  
Drug 3: $Y_{ij} = \beta_0+\beta_1 Time_{ij} + b_{0i} + b_{1i} Time_{ij} + e_{ij}$  

```{r}
dataLDA <- data.frame(id = Data3$MonkeyID, binding = Data3$Binding, Time_Point = Data3$Time_Point, D1 = as.numeric(Data3$Drug == 1), D2 = as.numeric(Data3$Drug == 2))
# head(dataLDA)

lda <- lme(fixed = binding ~ Time_Point + D1 + D1:Time_Point + D2 + D2:Time_Point,
           random = ~ Time_Point | id, weights = varIdent(form = ~ 1 | Time_Point), data = dataLDA, method = "REML")
summary(lda)
```

The p-values for `Drug` and the interaction of `Drug` and `Time_Point` are large. So we try another model with `Time_Point` as the only predictor.  [This is skipping the part where we fit only main effect (not interaction) with Drug]

$$Y_{ij} = \beta_0+ \beta_1 Time_{ij} + b_{0i} + b_{1i} Time_{ij} + e_{ij}$$  

$$
\underbrace{\begin{bmatrix}
Y_{i1} \\
\vdots \\
Y_{im_i}
\end{bmatrix}}_{\mathbf{Y_i}}
=
\underbrace{\begin{bmatrix}
1 & Time_{i1} \\
\vdots & \vdots & \vdots \\
1 & Time_{im_i} & 
\end{bmatrix}}_{\mathbf{X_i}}
\underbrace{\begin{bmatrix}
\beta_0 \\
\beta_1 
\end{bmatrix}}_{\mathbf{\beta}}
+
\underbrace{\begin{bmatrix}
1 & Time_{i1} \\
\vdots & \vdots \\
1 & Time_{im_i}
\end{bmatrix}}_{\mathbf{Z_i}} 
\underbrace{\begin{bmatrix}
b_{0i} \\
b_{1i}
\end{bmatrix}}_{\mathbf{b_i}}
+ 
\underbrace{\begin{bmatrix}
e_{i1} \\
\vdots \\
e_{im_i}
\end{bmatrix}}_{\mathbf{e_i}}$$

$$\mathbf{b_i} \sim N \left(0, \mathbf{D}=
\begin{bmatrix}
D_{11} & D_{12} \\
D_{12} & D_{22} 
\end{bmatrix}
\right)$$  

$$\mathbf{e}_{ij} \sim N(0, \mathbf{R}_i = \sigma^2 I_{mi})$$

[need to consider whether time point 2 is the optimal point]

```{r}
lda2 <- lme(fixed = binding ~ Time_Point,
           random = ~ Time_Point | id, weights = varIdent(form = ~ 1 | Time_Point), data = dataLDA, method = "REML")
summary(lda2)
```

This simpler model has lower AIC and BIC, as shown below. So we prefer the model with `Time_Point` as the predictor and, with the low p-values of the slope of `Time_Point`, conclude that the binding rates vary over time. In other words, the number of HIV vaccines given do affect the binding rate, but the drugs given do not have significant effects.  

```{r}
data.frame(cbind(AIC(lda, lda2), BIC(lda, lda2)))
```


# Discussion  

In this study we used both multivariate and longitudinal data analysis to examine the effects of HIV vaccines and Treg suppression treatments. Although the study provides evidence to support the concept of using immunosuppressing treatments to increase diversity, the added diversity does not seem to improve the binding rate. In other words, this study does not provide evidence to show that the added treatments can enhance the effects of HIV vaccines. 

# List of variables  

* Treatment: Treatment A is the mock control, and treatment B and C are two different kinds of Treg inhibitor treatments.
* Time Points: 0 represents before immunization; 1 represents 2 weeks post 1st immunization; 2 represents 2 weeks post 2nd immunization; and 3 represents 2 weeks post 3rd immunization, respectively.
* Isotype: the category of antibody type
* H_ID and L_ID: heavy chain and light chain IDs for the particular observation
* H_V, H_D and H_J: the gene segments used in heavy chain VDJ recombination in that antibody. The same applies to L_V and L_J
* H_VBase: the number of nucleotide of the heavy chain variable region
* H_Substitutions, H_Insertions, H_Deletions: the number of relative nucleotide mutations.
* HMuFreq: calculated by H_Substitutions / H_VBase
* H_CDR3: the number of amino acid of the heavy chainâ€™s third complementarity determining region 
* Binding: affinity of antibodies against a selected HIV glycoprotein. The larger value indicates stronger binding  

# Reference  

The dataset, [which can be found here](https://github.com/luokan1227/537P1), was provided by Kan Luo, as he was one of authors for the following four publications that used the dataset:  

1. Luo K, Liao HX, Zhang R, et al. Tissue memory B cell repertoire analysis after ALVAC/AIDSVAX B/E gp120 immunization of rhesus macaques. *JCI Insight*. 2016;1(20):e88522. Published 2016 Dec 8. doi:10.1172/jci.insight.88522
2. Bradley, T., Kuraoka, M., Yeh, C.-H., Tian, M., Chen, H., Cain, D. W., . . . Haynes, B. F. (2020). Immune checkpoint modulation enhances HIV-1 antibody induction. *Nature Communications*, 11(1), 948. doi:10.1038/s41467-020-14670-w
3. Easterhoff, D., Pollara, J., Luo, K., Tolbert, W. D., Young, B., Mielke, D., . . . Ferrari, G. (2020). Boosting with AIDSVAX B/E Enhances Env Constant Region 1 and 2 Antibody-Dependent Cellular Cytotoxicity Breadth and Potency. *Journal of Virology*, 94(4), e01120-01119. doi:10.1128/jvi.01120-19
4. Wiehe, K., Easterhoff, D., Luo, K., Nicely, N. I., Bradley, T., Jaeger, F. H., Dennison, S. M., Zhang, R., Lloyd, K. E., Stolarchuk, C., Parks, R., Sutherland, L. L., Scearce, R. M., Morris, L., Kaewkungwal, J., Nitayaphan, S., Pitisuttithum, P., Rerks-Ngarm, S., Sinangil, F., Phogat, S., â€¦ Haynes, B. F. (2014). Antibody light-chain-restricted recognition of the site of immune pressure in the RV144 HIV-1 vaccine trial is phylogenetically conserved. *Immunity*, 41(6), 909â€“918. https://doi.org/10.1016/j.immuni.2014.11.014  
5. Lefranc MP, Giudicelli V, Ginestoux C, Bodmer J, Muller W, Bontrop R, Lemaitre M, Malik A, Barbie V, Chaume D. IMGT, the international ImMunoGeneTics database. *Nucleic Acids Res*. 1999;27:209â€“212. doi: 10.1093/nar/27.1.209.  
6. Jenny M Woof , Dennis R Burton,Human antibody-Fc receptor interactions illuminated by crystal structures.*Nat Rev Immunol*. 2004 Feb;4(2):89-99. doi: 10.1038/nri1266.  
7. Pinheiro J, Bates D, DebRoy S, Sarkar D, R Core Team (2020). _nlme: Linear and Nonlinear Mixed Effects Models_. R package
version 3.1-150, <URL: https://CRAN.R-project.org/package=nlme>.

