---
title: "**Antibody Response Induced by HIV Vaccines and T-cell Suppression Treatments in Rhesus Macaques**"
author: "Group 3: Kan Luo, Shih-Ni Prim, Frederick Davey, Rizwana Rehman"
date: "11/12/2020"
fontsize: 12pt
geometry: margin = 1in
mainfont: Times New Roman
spacing: 1.5
fig_caption: yes
indent: true
header-includes:
  - \usepackage{endfloat}  
output: 
  bookdown::pdf_document2:
    latex_engine: lualatex
    toc: true
    fig_width: 6.5
    fig_height: 4
    df_print: kable
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, eval = TRUE, message = FALSE, warning = FALSE)
library(readxl)
library(tidyverse)
library(ggplot2)
library(httr)
library(nlme)
library(emmeans)
library(GGally)
library(knitr)
library(ICSNP)
library(bookdown)
```



```{r readin Data, results = 'hide'}
#Readin data from Github url address
url1 <- 'https://github.com/luokan1227/537P1/raw/master/Data.xlsx'  # main data
url2 <- 'https://github.com/luokan1227/537P1/raw/master/MonkeyID.xlsx'  # Monkey ID
GET(url1, write_disk(tf1 <- tempfile(fileext = ".xlsx")))
GET(url2, write_disk(tf2 <- tempfile(fileext = ".xlsx")))
Rawdata <- read_excel(tf1)
MonkeyID <-  read_excel(tf2)
#Add Monkey ID to raw data
Rawdata <- inner_join(Rawdata, MonkeyID, by = "H_ID")
colnames(Rawdata)[2] <- "Time_Point"
#-----------------------
#Adjust or add variables
#-----------------------
# Final data set for all following analysis
Data <- Rawdata
# add drug type and reactivity
Data$Drug <- ifelse(Data$Treatment == "group 7", 3, 
                      ifelse(Data$Treatment == "group 1", 1, 
                      ifelse(Data$Treatment == "group 2", 1, 
                      ifelse(Data$Treatment == "group 3", 1, 2))))
Data$Reactivity <- ifelse(Data$Binding > 0.1, 1, 0)
# create a subset with the variables we need plus the extracted information
Data2 <- Data %>% select(MonkeyID, Drug, Treatment, Time_Point, Isotype, H_VBase, H_Substitutions, H_Insertions, H_Deletions, HMuFreq, H_CDR3, L_VBase, L_Substitutions, L_Insertions, L_Deletions, LMuFreq, L_CDR3, Binding, Reactivity)
```

# Introduction  

## About the Study  

A dominant vaccine development strategy is to induce neutralizing antibodies by immunizing humans with the virus' glycoproteins. However, HIV vaccines that adopted this strategy mostly failed due to the fact that HIV is an RNA virus, which mutates rapidly to escape the inhibition of neutralizing antibodies. By the time the body generates neutralizing antibodies against the glycoproteins of some HIV strains, the RNA virus has already mutated. Thus, the existing neutralizing antibody fails to recognize, bind with, and neutralize the HIV virus. One possible solution is to increase the number of potential neutralizing antibodies that will cycle in the body by releasing a variety of antibodies after glycoprotein immunization.  

During the experiment, 20 rhesus macaques were given glycoprotein immunization and supplemental antibody doses, as well as one of three treatments (two experimental regulatory T-cell suppression treatments and one control). For the analysis of mutation frequency and CDR3 count, each antibody within the same treatment was treated as an independent observation.  

Regulatory T (Treg) cells prevent autoimmune diseases and suppress allergic reactions by inhibiting adaptive antibody immune response in the germinal center. Theoretically, this adaptive response lowers the effectiveness of vaccines. Thus the experiment used T-cell suppression treatments to investigate  the effect on immunization. These drugs are widely used in post transplant immuno-suppression treatment to prevent rejection.  

While we might expect different variance within subject vs between subjects, the number of potential antibodies observed is much higher than the number sampled in a blood draw. 

## About the Dataset  

Our dataset includes measurements of antibodies measured in 20 rhesus macaques after they were given the same HIV vaccine at three different time-points and one of three randomly selected anti-Treg treatments(drugs). Blood samples were collected two weeks after vaccine dosing, and antibodies were isolated from those samples. A different number of antibodies were collected from each blood sample, limited by assay yield. Each observation contains information about the antibody isolated post the glycoprotein immunization.  

A human antibody is formed by a heavy chain and light chain. For heavy chain, a human has about 51 V-gene segments, 25 D-gene segments and 6 J-gene segment. For light chain there are 71 V-gene segments and 9 J-gene segments[ref.5]. Any heavy chain V-D-J combination and light chain V-J combinations can randomly happen in germline center. Theoretically, there can be $51*25*6*71*9=$ `r 51*25*6*71*9` combinations of gene segments. Considering the frequently happened mutation and other factors, each individual can have over **10 billion** different antibodies. Thus, we decided to follow the convention of vaccine studies and treat each antibody as independent.   [Kan -- can you identify a reference article or journal here that uses this convention? We don't need to quote it / change the answer, it's just defending our claim that it's a standard practice]  

Below is the list of variables with a brief description from our dataset. Please note that in each antibody, there are two sets of heavy chain and light chain, all of which forming a Y-shape immunoglobulin. Thus many of the variables start with H or L, indicating which chain the information comes from.  

## List of variables  

* Monkey_id: Lists the identity of monkey
* Treatment(Drug): Treatment A is the mock control, and treatment B and C are two different kinds of Treg inhibitor treatments.
* Time_Points: 0 represents before immunization; 1 represents 2 weeks post 1st immunization; 2 represents 2 weeks post 2nd immunization; and 3 represents 2 weeks post 3rd immunization, respectively.
* Isotype: The category of antibody type; there are 5 kinds of immunoglobulin isotypes: IgG, IgA, IgM, IgE, IgD. The two most important kinds are IgG and IgM. IgM occurs in the acute stage of infection and perform an role of primary response. The secondary response IgG appears later in serum with higher binding affinity, and neutralizing potentials against toxins and virus. IgA mostly found in mucosal tissues such as Nasal mucosa. Non-dominant IgD and IgE are typically lower than 1% in blood. 
* H_ID and L_ID: heavy chain and light chain IDs for the particular observation
* H_VBase: the number of nucleotide of the heavy chain variable region
* H_Substitutions: the number of relative nucleotide mutations in heany chain.
* HMuFreq: calculated by H_Substitutions / H_VBase
* H_CDR3: the number of amino acid of the heavy chain's third complementarity determining region 
* L_VBase: the number of nucleotide of the light chain variable region
* L_Substitutions: the number of relative nucleotide mutations in light chain.
* LMuFreq: calculated by L_Substitutions / L_VBase
* L_CDR3: the number of amino acid of the light chain's third complementarity determining region.`H_CDR3` and `L_CDR3` indicates the length of the third complementarity-determining region on the variable heavy chain and light chain. The longer they are, the more potential there is to produce diverse antibodies. [Kan, could you check to see if this is correct?] In other words, we want the values to be higher. 
* Binding: affinity of antibodies against a selected HIV glycoprotein. The larger value indicates stronger binding.`Binding` indicates the rate of neutralizing, meaning how much the antibodies bind with the virus and thus make the virus ineffective. This is the most important measure of the study.   

## Research questions  

In the current report, we test if the different Treg treatments and number of vaccine injections(? which variables indicates this) cause changes in the antibody characteristics and if the changes are related to immunization/treatment time points.  

Specifically, we evaluate:  

$Q1$. Do the paired antibody heavy chain and light chain differ with respect to  mutation and CDR3?  
$Q2$. Do treatments, and/or isotypes have effects on the mutation frequency and/or the amino acid count in the third complementarity determining region (CDR3)?
  
$Q3$. How does the binding strength of the antibodies develop in response to the number of vaccine dosages by treatment? This will be evaluated with a longitudinal analysis by test subject.[NEEDS TO BE UPDATED]  

# Methods  

For exploratory data analysis we used histograms and Q-Q plots. We also computed correlations among variables. 

## Data Summaries  

The dataset has `r nrow(Data2)` antibodies collected from 20 rhesus monkeys. We first present our exploratory data analysis and summaries. An overview of the number of antibodies collected per treatment at each time-point is given below (Figure \@ref(fig:hist-antibodies)) along with tables for treatment counts per time-point and Isoptype per treatment for individual time-points.  

```{r}
Data2Summary<- Data2 %>% group_by(Drug, Time_Point, MonkeyID) %>% summarize(nObs = n())
Data2Summary$Drug<- as.factor(Data2Summary$Drug)
```

```{r, hist-antibodies, fig.cap = "Histograms of Antibodies"}
# "Histograms of Antibodies Collected per Treatment and Timepoint"
ggplot(data = Data2Summary, mapping = aes(x = nObs, color = Drug)) + geom_histogram(bins = 8, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + labs(xlab = 'n Antibodies Collected', ylab = 'n Monkeys')
```

```{r, ft-drug-timepoint}
knitr::kable(table(Data2$Drug, Data2$Time_Point, dnn= c('Drug', 'TimePoint')), caption = "Frequency tables of drug vs. timepoints")
```
```{r, ft-drug-timepoint-isotype}
knitr::kable(table(Data2$Drug, Data2$Time_Point, Data2$Isotype, dnn= c('Drug', 'TimePoint', 'Isotype')), caption = "Frequency tables of drug vs. timepoints vs. isotypes")
```

Next,  the histogram of  `Isotype` is presented in Figure \@ref(fig:isotype-plot).  

```{r, isotype-plot, fig.cap = "Histograms of Isotypes"}
# Histogram 
Iso.order <- c("G", "M", "A", "D", "E") 
g1 <- ggplot(Data, aes(x=factor(Isotype, levels = Iso.order), fill=as.factor(Drug))) 
  g1 + geom_bar(stat = "count", width = .9, position = 'Stack') + # his plot
    facet_grid(cols = vars(Time_Point), labeller = label_both) +
    labs(fill = 'Drug') + # Legend Content
    xlab("Ig Isotype") + ylab("Number of abs") +# X axis and y axis title
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 20) #Title
          #legend.text = element_text(size=20), #Legend text
          #legend.title = element_text(size = 20) # Legend title
    )
```

As expected, we observed that IgG and IgM occupied the biggest proportion of all antibodies in all time points. Before immunization (time point 0), there were similar weight of IgG and IgM found in blood. After the 1st immunization (time point 1), primary immune response resulted an increase of IgM, followed with IgG increase at later time point 2 and 3. 

For the response variables, we began with an outlier check, and then reviewed their distributions across treatment and time-points.

```{r Outlier Review}
tibContinuous <- gather(Data2, key = "Variable", value = "value", 
                        H_CDR3, HMuFreq, L_CDR3, LMuFreq, Binding)
#histograms of continuous variables
ggplot(data = tibContinuous, aes(x = value)) + 
  geom_histogram(bins = 20) +
  facet_wrap(~Variable, scales = 'free') + labs(title = 'Figure 3: Histogram of Response Variables')
t<- summary(scale(Data2$L_CDR3))
kable(t(t), caption = 'Normal Standardized L_CDR3 Statistics')
```
L_CDR3 appeared to have an outlier several standard (?distribution).  We believed that this  most likely was caused by a laboratory error, so we excluded this observation  and continued the analysis with the remaining data of 2464 observations.

```{r}
Data3 <- Data2[-972,]
```

We examined our responses: `H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq` and `Binding`. 

We observed that for `H_CDR3` the distributions were roughly normal with the center around 13 at different time-points (Figure 4) without taking into account different treatments. Figure 5 represents the distribution of `H_CDR3` with respect to treatments at different time-points.  , and slightly centered around 9 for `L_CDR3` at different time- points (Figure 5).  

```{r H_CDR3 Plot/Table, echo=FALSE}
g3.1 <- ggplot(Data3, aes(H_CDR3))
  g3.1 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    ggtitle("Figure 4: Histogram H_CDR3") +  #Plot title
    xlab("HCDR3 Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    scale_x_continuous(breaks = round(seq(min(Data$H_CDR3), max(Data$H_CDR3), by = 1),1)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18))  #Title
  
  
  g4<- ggplot(data = Data3, mapping = aes(x = H_CDR3, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'H_CDR3', ylab = 'Count', title= "Figure 5: Histograms of H_CDR3 vs. Treatment and Timepoint")
g4 

```

Figures for `L_CDR3` are shown next. 

```{r L_CDR3 Plot/Table, echo=FALSE}
g5 <- ggplot(Data3, aes(L_CDR3))
  g5 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    ggtitle("Figure 6: Histogram L_CDR3") +  #Plot title
    xlab("LCDR3 Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    scale_x_continuous(breaks = round(seq(min(Data$L_CDR3), max(Data$L_CDR3), by = 1),1)) + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )

g6<- ggplot(data = Data3, mapping = aes(x = L_CDR3, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'L_CDR3', ylab = 'Count', title= 'Figure 7: Histograms of L_CDR3 vs. Treatment and Timepoint')
g6
```

Q-Q plots of `H_CDR3` and `L_CDR3` are shown below.

```{r QQ, echo=FALSE, warning=FALSE}
#par(mfrow = c(1,2))
qqnorm(Data3$H_CDR3, main = "Figure 8a: Q-Q Plot of H_CDR3", pch = 19, cex = 1)
qqnorm(Data3$L_CDR3, main = "Figure 8b: Q-Q Plot of L_CDR3", pch = 19, cex = 1)
```
`HMuFreq` and `LMuFreq` were calculated by dividing `H_Substitution` by `H_VBase` for heavy chain and similarly for light chain. These two variables show how much the antibodies mutate. A higher mutation rate is usually indicative of better virus neutralization. Below we present comparison of mutation rate between heavy chain and light chain. 

```{r mutation rate, echo=FALSE, warning=FALSE}
#par(mfrow = c(1,2))
g7 <- ggplot(Data3, aes(HMuFreq))
  g7 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    ggtitle("Figure 9: Histogram HMuFreq") +  #Plot title
    xlab("HMuFreq Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )
g8<- ggplot(data = Data3, mapping = aes(x = HMuFreq, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'HMuFreq', ylab = 'n Monkeys', title= 'Figure 10: Histograms of HMuFreq vs. Treatment and Timepoint')
g8

g9 <- ggplot(Data3, aes(LMuFreq))
  g9 + geom_bar(stat = "count", width = 0.7, fill = "#012169") +
    facet_grid(Time_Point~., labeller = label_both) + #by subgroups
    ggtitle("Figure 11: Histogram LMuFreq") +  #Plot title
    xlab("LMuFreq Length (aa)") + ylab("Number of antibodies") +# X axis and y axis title
    
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),  #remove background grid
          panel.background = element_blank(),  #remove background color
          axis.text.x = element_text(size = 10), # x axis 
          axis.text.y = element_text(size = 10), # y axis 
          axis.title.x = element_text(size = 15), # x title 
          axis.title.y = element_text(size = 15), # y title
          plot.title = element_text(hjust = 0.5, size = 18) #Title
          )

g10<- ggplot(data = Data3, mapping = aes(x = LMuFreq, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white' ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'LMuFreq', ylab = 'n Monkeys', title= 'Figure 12: Histograms of LMuFreq vs Treatment and Timepoint')
g10
```
Next, the Q-Q plots of `HMuFreq` and `LMuFreq` are shown.

```{r HL QQ, echo=FALSE, warning=FALSE}
qqnorm(Data3$HMuFreq, main = "Figure 13a: Q-Q Plot of HMuFreq", pch = 19, cex = 1)
qqnorm(Data3$LMuFreq, main = "Figure 13b: Q-Q Plot of LMuFreq", pch = 19, cex = 1)
```

Lastly, Histogram of `Binding` with respect to treatments at different time points and Q-Q plot are shown. We observed that `Binding` was not normally distributed. However, since our sample size is larger than 2000, we can use the Central Limit Theorem and assume normality.  

```{r}

g8<- ggplot(data = Data3, mapping = aes(x = Binding, color = as.factor(Drug))) + 
  geom_histogram(bins = 10, fill = 'white', aes(y=..density..) ) + facet_grid(Drug~Time_Point, labeller = label_both) + 
  labs(xlab = 'Binding', ylab = 'Count', title= 'Figure 14: Histograms of Binding Strength vs Treatment and Timepoint')
g8
qqnorm(Data3$Binding, main="Figure 15: Q-Q Plot of Binding", pch = 19, cex = 1)
```
We checked whether response variables could be correlated. In the plot below, we observed that none of the response variables were highly correlated.

```{r Q_Q binding}
Data3 %>% select(LMuFreq, HMuFreq, L_CDR3, H_CDR3, Binding) %>% ggpairs()
```

## Multivariate Data Analysis  

For inferential analysis we used Dependent Hotellings $T^2$ test using the ICSNP package [xxx] for Q1. To answer Q2 $i.e.,$ to evaluate the mutation frequency and CDR count vs. Treatments and Isotypes we used two-way MANOVA.  we ran the [xxx] function from the [xxx] package [package citation]. It was followed by further pair-wise analysis. 

To answer $Q1$ we conducted  Hotellings $T^2$ test for a  pairwise comparison between respective heavy chain and light chain variable pairs `L_CDR3`, `H_CDR3` and `HMuFreq`, `LMuFreq`.

```{r hotelling }
Light <- data.frame(Data3$L_CDR3,Data3$LMuFreq)
Heavy <- data.frame(Data3$H_CDR3,Data3$HMuFreq)
# paired test
D <- Light - Heavy
HotellingsT2(D, na.action=na.exclude)
```

For $Q2$ We wanted to test whether predictors `Drug` and `Isotype` had effects on the five responses: `H_CDR3`, `HMuFreq`, `L_CDR3`, `LMuFreq`, and `Binding`. We used $MANOVA$ due to large sample size (n = 2464) without worrying about the normality assumption.   In the output, we noticed that the main effects of `Drug` and `Isotype` and the interaction effects were all significant.  



```{r}
ID <- as.factor(Data3$MonkeyID)
trt <- as.factor(Data3$Treatment)
drug <- as.factor(Data3$Drug)
tp <- as.factor(Data3$Time_Point)
it <- as.factor(Data3$Isotype)
fit.manova <- manova(cbind(Data3$H_CDR3, Data3$HMuFreq, Data3$L_CDR3, Data3$LMuFreq, Data3$Binding) ~ drug*it)
summary(fit.manova)
```
### Pairwise comparison

To know more details about which groups have different means, we use pairwise comparisons for each treatment group, drug, and Isotype. (We set $\alpha$ as 0.05, and use a Bonferroni correction.)  

```{r}
respMat <- as.matrix(Data3[,c("L_CDR3", "LMuFreq", "H_CDR3", "HMuFreq", "Binding")])
# pairwise comparison among treatment groups
vars <- c("L_CDR3", "LMuFreq", "H_CDR3", "HMuFreq", "Binding")
```

```{r}
# pairwise comparison among drug groups
fit2 <- manova(respMat[,1:5] ~ drug)
# summary(fit2)
p <- 5
q2 <- length(unique(drug))
alpha.old <- 0.05
nc2 <- p*q2*(q2-1)/2
alpha.new2 <- alpha.old/nc2
for (i in 1:5){
  w <- c(0, 0, 0, 0, 0)
  w[i] <- 1
  print(paste(vars[i], " pairwise CI's"))
  cont <- contrast(emmeans(fit2, "drug", weights = w), "pairwise")
  bb <- confint(cont, level = 1 - alpha.new2, adj = "none")
  print(bb)
}
```

```{r}
# pairwise comparison among time point
fit3 <- manova(respMat[,1:5] ~ it)
# summary(fit3)
p <- 5
q3 <- length(unique(it))
alpha.old <- 0.05
nc3 <- p*q3*(q3-1)/2
alpha.new3 <- alpha.old/nc3
for (i in 1:5){
  w <- c(0, 0, 0, 0, 0)
  w[i] <- 1
  print(paste(vars[i], " pairwise CI's"))
  cont <- contrast(emmeans(fit3, "it", weights = w), "pairwise")
  bb <- confint(cont, level = 1 - alpha.new3, adj = "none")
  print(bb)
}
```

Here are the pairs that have significant differences:  

* Treatment
    * L_CDR3: none
    * LMuFreq: none
    * H_CDR3: 
        * group 1 > group 4
    * HMuFreq: 
        * group 1 > group 7
        * group 3 < group 5
        * group 3 < group 6
        * group 5 > group 7
        * group 6 > group 7
    * Binding:
        * group 1 < group 2
        * group 1 > group 6
        * group 2 > group 3
        * group 2 > group 4
        * group 2 > group 5
        * group 2 > group 6
        * group 3 < group 7
        * group 4 > group 6
        * group 4 < group 7
        * group 5 < group 7
        * group 6 < group 7
* Drug
    * L_CDR3: none
    * LMuFreq: none
    * H_CDR3: 
        * 1 > 2
    * HMuFreq: 
        * 1 < 2
        * 1 > 3
        * 2 > 3
    * Binding: 
        * 1 < 3
        * 2 < 3
* Isotype
    * L_CDR3: none
    * LMuFreq: none
    * H_CDR3: 
        * IgD < IgG
    * HMuFreq: 
        * IgD < IgG
        * IgG > IgM
    * Binding: 
        * IgA < IgG
        * IgD < IgG
        * IgG > IgM

In short, `L_CDR3` and `LMuFreq` do not have significnat paired differences.  

For `H_CDR3`, treatment group 1 (drug 1) is higher than treatment group 7 (control), and IgG has a longer H_CDR3 length than IgD.  

For `HMuFreq`, drug 2 has the highest mean, followed by drug 1 and control. More specifically, treatment groups 5 and 6 (two doses in drug 2) have the highest mutation rates. IgG has higher mutation rate than IgD.  

For `Binding`, drug 3 has the highest mean, but drug 1 and drug 2 do not have significant differences. IgG has higher binding rate than IgA, IgD, and IgM.  

We could conclude that the drugs/treatment groups do increase mutation rate; however, they do not increase binding rate. That is to say, although the treatments do help increase the diversity of antibodies, they are not specific to the HIV antigens and thus do not increase binding. 

## Longitudinal Analysis  

For the longitudinal analysis of binding strength vs number of vaccine doses, we will be using the gls function from the nlme package$^{[7]}.$ (NEEDS TO UP DATED)

# Results  

# Discussion  
**Test Assumptions**
**Results**
**Implications (Interpretation for practical significance**)

# Conclusions  

# References  

The dataset, [which can be found here](https://github.com/luokan1227/537P1), was provided by Kan Luo, as he was one of authors for the following four publications that used the dataset:  

1. Luo K, Liao HX, Zhang R, et al. Tissue memory B cell repertoire analysis after ALVAC/AIDSVAX B/E gp120 immunization of rhesus macaques. *JCI Insight*. 2016;1(20):e88522. Published 2016 Dec 8. doi:10.1172/jci.insight.88522
2. Bradley, T., Kuraoka, M., Yeh, C.-H., Tian, M., Chen, H., Cain, D. W., . . . Haynes, B. F. (2020). Immune checkpoint modulation enhances HIV-1 antibody induction. *Nature Communications*, 11(1), 948. doi:10.1038/s41467-020-14670-w
3. Easterhoff, D., Pollara, J., Luo, K., Tolbert, W. D., Young, B., Mielke, D., . . . Ferrari, G. (2020). Boosting with AIDSVAX B/E Enhances Env Constant Region 1 and 2 Antibody-Dependent Cellular Cytotoxicity Breadth and Potency. *Journal of Virology*, 94(4), e01120-01119. doi:10.1128/jvi.01120-19
4. Wiehe, K., Easterhoff, D., Luo, K., Nicely, N. I., Bradley, T., Jaeger, F. H., Dennison, S. M., Zhang, R., Lloyd, K. E., Stolarchuk, C., Parks, R., Sutherland, L. L., Scearce, R. M., Morris, L., Kaewkungwal, J., Nitayaphan, S., Pitisuttithum, P., Rerks-Ngarm, S., Sinangil, F., Phogat, S., . Haynes, B. F. (2014). Antibody light-chain-restricted recognition of the site of immune pressure in the RV144 HIV-1 vaccine trial is phylogenetically conserved. *Immunity*, 41(6), 909-918. https://doi.org/10.1016/j.immuni.2014.11.014  
5. Lefranc MP, Giudicelli V, Ginestoux C, Bodmer J, Muller W, Bontrop R, Lemaitre M, Malik A, Barbie V, Chaume D. IMGT, the international ImMunoGeneTics database. *Nucleic Acids Res*. 1999;27:209-212. doi: 10.1093/nar/27.1.209.  
6. Jenny M Woof , Dennis R Burton,Human antibody-Fc receptor interactions illuminated by crystal structures.*Nat Rev Immunol*. 2004 Feb;4(2):89-99. doi: 10.1038/nri1266.  
7. Pinheiro J, Bates D, DebRoy S, Sarkar D, R Core Team (2020). _nlme: Linear and Nonlinear Mixed Effects Models_. R package
